{
  "cells": [
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "view-in-github",
        "colab_type": "text"
      },
      "source": [
        "<a href=\"https://colab.research.google.com/github/Vladm0z/hse_hw2_chip/blob/main/hse_hw2_chip.ipynb\" target=\"_parent\"><img src=\"https://colab.research.google.com/assets/colab-badge.svg\" alt=\"Open In Colab\"/></a>"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "XUZS4GUkHIEG"
      },
      "source": [
        "# Подготовка файлов и программ"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "id": "lrtRbP3fAvj8"
      },
      "outputs": [],
      "source": [
        "!pip install -q condacolab"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "id": "jQ1Oqzb9BXoG"
      },
      "outputs": [],
      "source": [
        "import condacolab"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "q5FUOHCgBZWp",
        "outputId": "2d8835cc-44c9-4b52-dc67-ee7173b3909d"
      },
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "✨🍰✨ Everything looks OK!\n"
          ]
        }
      ],
      "source": [
        "condacolab.install()"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "id": "XnbIt77YBOT9",
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "outputId": "a1d58882-e9da-4fe1-b133-8f68efaf067a"
      },
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Collecting package metadata (current_repodata.json): - \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\bdone\n",
            "Solving environment: / \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\bdone\n",
            "\n",
            "# All requested packages already installed.\n",
            "\n"
          ]
        }
      ],
      "source": [
        "!conda install -c bioconda seqtk"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "sCBV_31qHOik",
        "outputId": "cfe59e38-1aa5-4537-fd0e-5454ecb1a602"
      },
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Collecting package metadata (current_repodata.json): - \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\bdone\n",
            "Solving environment: \\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\bdone\n",
            "\n",
            "## Package Plan ##\n",
            "\n",
            "  environment location: /usr/local\n",
            "\n",
            "  added / updated specs:\n",
            "    - trimmomatic\n",
            "\n",
            "\n",
            "The following packages will be downloaded:\n",
            "\n",
            "    package                    |            build\n",
            "    ---------------------------|-----------------\n",
            "    openjdk-8.0.312            |       h7f98852_0        97.6 MB  conda-forge\n",
            "    python_abi-3.7             |          2_cp37m           4 KB  conda-forge\n",
            "    trimmomatic-0.39           |       hdfd78af_2         144 KB  bioconda\n",
            "    ------------------------------------------------------------\n",
            "                                           Total:        97.8 MB\n",
            "\n",
            "The following NEW packages will be INSTALLED:\n",
            "\n",
            "  openjdk            conda-forge/linux-64::openjdk-8.0.312-h7f98852_0\n",
            "  trimmomatic        bioconda/noarch::trimmomatic-0.39-hdfd78af_2\n",
            "\n",
            "The following packages will be UPDATED:\n",
            "\n",
            "  python_abi                                    3.7-1_cp37m --> 3.7-2_cp37m\n",
            "\n",
            "\n",
            "\n",
            "Downloading and Extracting Packages\n",
            "trimmomatic-0.39     | 144 KB    | : 100% 1.0/1 [00:00<00:00,  7.76it/s]\n",
            "openjdk-8.0.312      | 97.6 MB   | : 100% 1.0/1 [00:11<00:00, 11.09s/it]               \n",
            "python_abi-3.7       | 4 KB      | : 100% 1.0/1 [00:00<00:00, 32.12it/s]\n",
            "Preparing transaction: / \b\bdone\n",
            "Verifying transaction: \\ \b\bdone\n",
            "Executing transaction: / \b\bdone\n"
          ]
        }
      ],
      "source": [
        "!conda install -c bioconda trimmomatic"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 1000
        },
        "id": "F0BLWsH2n3Ov",
        "outputId": "ec7c4c26-4d64-43fc-b2ee-e6030207a588"
      },
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Collecting macs2\n",
            "  Downloading MACS2-2.2.7.1.tar.gz (2.1 MB)\n",
            "\u001b[K     |████████████████████████████████| 2.1 MB 16.1 MB/s \n",
            "\u001b[?25h  Installing build dependencies ... \u001b[?25l\u001b[?25hdone\n",
            "  Getting requirements to build wheel ... \u001b[?25l\u001b[?25hdone\n",
            "    Preparing wheel metadata ... \u001b[?25l\u001b[?25hdone\n",
            "Collecting intervene\n",
            "  Downloading intervene-0.6.5.tar.gz (5.5 MB)\n",
            "\u001b[K     |████████████████████████████████| 5.5 MB 29.8 MB/s \n",
            "\u001b[?25hCollecting pybedtools\n",
            "  Downloading pybedtools-0.9.0.tar.gz (12.5 MB)\n",
            "\u001b[K     |████████████████████████████████| 12.5 MB 62.6 MB/s \n",
            "\u001b[?25hCollecting matplotlib\n",
            "  Downloading matplotlib-3.5.1-cp37-cp37m-manylinux_2_5_x86_64.manylinux1_x86_64.whl (11.2 MB)\n",
            "\u001b[K     |████████████████████████████████| 11.2 MB 71.6 MB/s \n",
            "\u001b[?25hCollecting pandas\n",
            "  Downloading pandas-1.3.5-cp37-cp37m-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (11.3 MB)\n",
            "\u001b[K     |████████████████████████████████| 11.3 MB 54.9 MB/s \n",
            "\u001b[?25hCollecting numpy\n",
            "  Using cached numpy-1.21.5-cp37-cp37m-manylinux_2_12_x86_64.manylinux2010_x86_64.whl (15.7 MB)\n",
            "Collecting scipy\n",
            "  Downloading scipy-1.7.3-cp37-cp37m-manylinux_2_12_x86_64.manylinux2010_x86_64.whl (38.1 MB)\n",
            "\u001b[K     |████████████████████████████████| 38.1 MB 1.2 MB/s \n",
            "\u001b[?25hCollecting seaborn\n",
            "  Downloading seaborn-0.11.2-py3-none-any.whl (292 kB)\n",
            "\u001b[K     |████████████████████████████████| 292 kB 72.9 MB/s \n",
            "\u001b[?25hCollecting fonttools>=4.22.0\n",
            "  Downloading fonttools-4.29.1-py3-none-any.whl (895 kB)\n",
            "\u001b[K     |████████████████████████████████| 895 kB 64.4 MB/s \n",
            "\u001b[?25hCollecting packaging>=20.0\n",
            "  Downloading packaging-21.3-py3-none-any.whl (40 kB)\n",
            "\u001b[K     |████████████████████████████████| 40 kB 4.8 MB/s \n",
            "\u001b[?25hCollecting pillow>=6.2.0\n",
            "  Downloading Pillow-9.0.1-cp37-cp37m-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (4.3 MB)\n",
            "\u001b[K     |████████████████████████████████| 4.3 MB 51.1 MB/s \n",
            "\u001b[?25hCollecting kiwisolver>=1.0.1\n",
            "  Downloading kiwisolver-1.3.2-cp37-cp37m-manylinux_2_5_x86_64.manylinux1_x86_64.whl (1.1 MB)\n",
            "\u001b[K     |████████████████████████████████| 1.1 MB 58.9 MB/s \n",
            "\u001b[?25hCollecting cycler>=0.10\n",
            "  Downloading cycler-0.11.0-py3-none-any.whl (6.4 kB)\n",
            "Collecting pyparsing>=2.2.1\n",
            "  Downloading pyparsing-3.0.7-py3-none-any.whl (98 kB)\n",
            "\u001b[K     |████████████████████████████████| 98 kB 7.4 MB/s \n",
            "\u001b[?25hCollecting python-dateutil>=2.7\n",
            "  Downloading python_dateutil-2.8.2-py2.py3-none-any.whl (247 kB)\n",
            "\u001b[K     |████████████████████████████████| 247 kB 76.5 MB/s \n",
            "\u001b[?25hRequirement already satisfied: six>=1.5 in /usr/local/lib/python3.7/site-packages (from python-dateutil>=2.7->matplotlib->intervene) (1.15.0)\n",
            "Collecting pytz>=2017.3\n",
            "  Downloading pytz-2021.3-py2.py3-none-any.whl (503 kB)\n",
            "\u001b[K     |████████████████████████████████| 503 kB 75.4 MB/s \n",
            "\u001b[?25hCollecting pysam\n",
            "  Downloading pysam-0.18.0-cp37-cp37m-manylinux_2_12_x86_64.manylinux2010_x86_64.whl (14.9 MB)\n",
            "\u001b[K     |████████████████████████████████| 14.9 MB 41.1 MB/s \n",
            "\u001b[?25hBuilding wheels for collected packages: intervene, macs2, pybedtools\n",
            "  Building wheel for intervene (setup.py) ... \u001b[?25l\u001b[?25hdone\n",
            "  Created wheel for intervene: filename=intervene-0.6.5-py3-none-any.whl size=5546118 sha256=5ce325067cc19d8a8e55fba7a4b12a8acbec38583d8c74343a23d4939c866054\n",
            "  Stored in directory: /root/.cache/pip/wheels/f3/d2/c1/18fa6122c36fdc731af4f16c59c356606b12a890a578294044\n",
            "  Building wheel for macs2 (PEP 517) ... \u001b[?25l\u001b[?25hdone\n",
            "  Created wheel for macs2: filename=MACS2-2.2.7.1-cp37-cp37m-linux_x86_64.whl size=1790967 sha256=10a280227db762a9e08aa99def503c37997de5da90ae6c56031e4f4b96260a6a\n",
            "  Stored in directory: /root/.cache/pip/wheels/4e/34/10/f6222945f7c5b091e34b058a01a842ba9b1d960f2ab0c8e16f\n",
            "  Building wheel for pybedtools (setup.py) ... \u001b[?25l\u001b[?25hdone\n",
            "  Created wheel for pybedtools: filename=pybedtools-0.9.0-cp37-cp37m-linux_x86_64.whl size=13654293 sha256=ffd795164f68278e0b46ad51877a87aedc1befeca6832d0e515b66f2b248ef79\n",
            "  Stored in directory: /root/.cache/pip/wheels/7a/44/0d/3a7449885adaf8ebb157da8c3c834a712f48b3b3b84ba51dda\n",
            "Successfully built intervene macs2 pybedtools\n",
            "Installing collected packages: pyparsing, pytz, python-dateutil, pillow, packaging, numpy, kiwisolver, fonttools, cycler, scipy, pysam, pandas, matplotlib, seaborn, pybedtools, macs2, intervene\n",
            "Successfully installed cycler-0.11.0 fonttools-4.29.1 intervene-0.6.5 kiwisolver-1.3.2 macs2-2.2.7.1 matplotlib-3.5.1 numpy-1.21.5 packaging-21.3 pandas-1.3.5 pillow-9.0.1 pybedtools-0.9.0 pyparsing-3.0.7 pysam-0.18.0 python-dateutil-2.8.2 pytz-2021.3 scipy-1.7.3 seaborn-0.11.2\n"
          ]
        },
        {
          "output_type": "display_data",
          "data": {
            "application/vnd.colab-display-data+json": {
              "pip_warning": {
                "packages": [
                  "cycler",
                  "dateutil",
                  "kiwisolver",
                  "numpy",
                  "packaging",
                  "pandas",
                  "pyparsing"
                ]
              }
            }
          },
          "metadata": {}
        }
      ],
      "source": [
        "!pip install macs2 intervene"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "Ce6hiSTdkarU",
        "outputId": "887bf202-2d90-484e-8858-f4e0efb71da2"
      },
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Reading package lists... Done\n",
            "Building dependency tree       \n",
            "Reading state information... Done\n",
            "The following package was automatically installed and is no longer required:\n",
            "  libnvidia-common-470\n",
            "Use 'apt autoremove' to remove it.\n",
            "The following NEW packages will be installed:\n",
            "  bedtools\n",
            "0 upgraded, 1 newly installed, 0 to remove and 39 not upgraded.\n",
            "Need to get 577 kB of archives.\n",
            "After this operation, 2,040 kB of additional disk space will be used.\n",
            "Get:1 http://archive.ubuntu.com/ubuntu bionic/universe amd64 bedtools amd64 2.26.0+dfsg-5 [577 kB]\n",
            "Fetched 577 kB in 0s (5,770 kB/s)\n",
            "Selecting previously unselected package bedtools.\n",
            "(Reading database ... 155320 files and directories currently installed.)\n",
            "Preparing to unpack .../bedtools_2.26.0+dfsg-5_amd64.deb ...\n",
            "Unpacking bedtools (2.26.0+dfsg-5) ...\n",
            "Setting up bedtools (2.26.0+dfsg-5) ...\n"
          ]
        }
      ],
      "source": [
        "!apt-get install bedtools"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "AUML3dvsqT9E",
        "outputId": "751c8459-ded2-4d4b-afbc-37f4944b3958"
      },
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Reading package lists... Done\n",
            "Building dependency tree       \n",
            "Reading state information... Done\n",
            "The following package was automatically installed and is no longer required:\n",
            "  libnvidia-common-470\n",
            "Use 'apt autoremove' to remove it.\n",
            "The following additional packages will be installed:\n",
            "  cwltool libc-ares2 libhts2 libjs-bootstrap nodejs nodejs-doc\n",
            "  python-asn1crypto python-avro python-cachecontrol python-certifi\n",
            "  python-cffi-backend python-chardet python-cryptography python-enum34\n",
            "  python-html5lib python-idna python-ipaddress python-isodate python-lockfile\n",
            "  python-mistune python-openssl python-pkg-resources python-pyparsing\n",
            "  python-rdflib python-rdflib-jsonld python-requests python-ruamel.yaml\n",
            "  python-schema-salad python-shellescape python-six python-sparqlwrapper\n",
            "  python-typing python-urllib3 python-webencodings\n",
            "Suggested packages:\n",
            "  bowtie2-examples python-cryptography-doc python-cryptography-vectors\n",
            "  python-enum34-doc python-genshi python-lxml python-lockfile-doc\n",
            "  python-openssl-doc python-openssl-dbg python-setuptools python-pyparsing-doc\n",
            "  python-rdflib-doc python-rdflib-tools python-socks python-ntlm\n",
            "The following NEW packages will be installed:\n",
            "  bowtie2 cwltool libc-ares2 libhts2 libjs-bootstrap nodejs nodejs-doc\n",
            "  python-asn1crypto python-avro python-cachecontrol python-certifi\n",
            "  python-cffi-backend python-chardet python-cryptography python-enum34\n",
            "  python-html5lib python-idna python-ipaddress python-isodate python-lockfile\n",
            "  python-mistune python-openssl python-pkg-resources python-pyparsing\n",
            "  python-rdflib python-rdflib-jsonld python-requests python-ruamel.yaml\n",
            "  python-schema-salad python-shellescape python-six python-sparqlwrapper\n",
            "  python-typing python-urllib3 python-webencodings samtools\n",
            "0 upgraded, 36 newly installed, 0 to remove and 39 not upgraded.\n",
            "Need to get 9,743 kB of archives.\n",
            "After this operation, 46.1 MB of additional disk space will be used.\n",
            "Get:1 http://archive.ubuntu.com/ubuntu bionic/main amd64 python-pkg-resources all 39.0.1-2 [128 kB]\n",
            "Get:2 http://archive.ubuntu.com/ubuntu bionic/main amd64 python-six all 1.11.0-2 [11.3 kB]\n",
            "Get:3 http://archive.ubuntu.com/ubuntu bionic/universe amd64 python-isodate all 0.6.0-1 [23.8 kB]\n",
            "Get:4 http://archive.ubuntu.com/ubuntu bionic/main amd64 python-pyparsing all 2.2.0+dfsg1-2 [52.1 kB]\n",
            "Get:5 http://archive.ubuntu.com/ubuntu bionic/universe amd64 python-rdflib all 4.2.1-2 [237 kB]\n",
            "Get:6 http://archive.ubuntu.com/ubuntu bionic/main amd64 python-certifi all 2018.1.18-2 [144 kB]\n",
            "Get:7 http://archive.ubuntu.com/ubuntu bionic/main amd64 python-chardet all 3.0.4-1 [80.3 kB]\n",
            "Get:8 http://archive.ubuntu.com/ubuntu bionic/main amd64 python-idna all 2.6-1 [32.4 kB]\n",
            "Get:9 http://archive.ubuntu.com/ubuntu bionic-updates/main amd64 python-urllib3 all 1.22-1ubuntu0.18.04.2 [86.0 kB]\n",
            "Get:10 http://archive.ubuntu.com/ubuntu bionic-updates/main amd64 python-requests all 2.18.4-2ubuntu0.1 [58.5 kB]\n",
            "Get:11 http://archive.ubuntu.com/ubuntu bionic/universe amd64 python-ruamel.yaml amd64 0.15.34-1 [183 kB]\n",
            "Get:12 http://archive.ubuntu.com/ubuntu bionic/universe amd64 python-cachecontrol all 0.11.7-1 [15.2 kB]\n",
            "Get:13 http://archive.ubuntu.com/ubuntu bionic/main amd64 python-lockfile all 1:0.12.2-2 [14.6 kB]\n",
            "Get:14 http://archive.ubuntu.com/ubuntu bionic/universe amd64 python-mistune all 0.8.3-2 [13.8 kB]\n",
            "Get:15 http://archive.ubuntu.com/ubuntu bionic/universe amd64 python-rdflib-jsonld all 0.4.0-3 [12.2 kB]\n",
            "Get:16 http://archive.ubuntu.com/ubuntu bionic/universe amd64 python-typing all 3.6.2-1 [22.2 kB]\n",
            "Get:17 http://archive.ubuntu.com/ubuntu bionic/universe amd64 python-avro all 1.8.2+dfsg-1 [23.2 kB]\n",
            "Get:18 http://archive.ubuntu.com/ubuntu bionic/universe amd64 libjs-bootstrap all 3.3.7+dfsg-2 [228 kB]\n",
            "Get:19 http://archive.ubuntu.com/ubuntu bionic/universe amd64 python-schema-salad all 2.6.20171201034858-3 [260 kB]\n",
            "Get:20 http://archive.ubuntu.com/ubuntu bionic/universe amd64 python-shellescape all 3.4.1-2 [4,432 B]\n",
            "Get:21 http://archive.ubuntu.com/ubuntu bionic/universe amd64 cwltool all 1.0.20180302231433-1 [143 kB]\n",
            "Get:22 http://archive.ubuntu.com/ubuntu bionic/universe amd64 libhts2 amd64 1.7-2 [300 kB]\n",
            "Get:23 http://archive.ubuntu.com/ubuntu bionic-updates/universe amd64 nodejs-doc all 8.10.0~dfsg-2ubuntu0.4 [752 kB]\n",
            "Get:24 http://archive.ubuntu.com/ubuntu bionic/main amd64 python-asn1crypto all 0.24.0-1 [72.7 kB]\n",
            "Get:25 http://archive.ubuntu.com/ubuntu bionic/main amd64 python-cffi-backend amd64 1.11.5-1 [63.4 kB]\n",
            "Get:26 http://archive.ubuntu.com/ubuntu bionic/main amd64 python-enum34 all 1.1.6-2 [34.8 kB]\n",
            "Get:27 http://archive.ubuntu.com/ubuntu bionic/main amd64 python-ipaddress all 1.0.17-1 [18.2 kB]\n",
            "Get:28 http://archive.ubuntu.com/ubuntu bionic-updates/main amd64 python-cryptography amd64 2.1.4-1ubuntu1.4 [276 kB]\n",
            "Get:29 http://archive.ubuntu.com/ubuntu bionic/main amd64 python-webencodings all 0.5-2 [10.3 kB]\n",
            "Get:30 http://archive.ubuntu.com/ubuntu bionic/main amd64 python-html5lib all 0.999999999-1 [83.6 kB]\n",
            "Get:31 http://archive.ubuntu.com/ubuntu bionic/main amd64 python-openssl all 17.5.0-1ubuntu1 [41.3 kB]\n",
            "Get:32 http://archive.ubuntu.com/ubuntu bionic/universe amd64 python-sparqlwrapper all 1.7.6-2 [21.8 kB]\n",
            "Get:33 http://archive.ubuntu.com/ubuntu bionic/universe amd64 samtools amd64 1.7-1 [288 kB]\n",
            "Get:34 http://archive.ubuntu.com/ubuntu bionic/universe amd64 bowtie2 amd64 2.3.4.1-1 [1,174 kB]\n",
            "Get:35 http://archive.ubuntu.com/ubuntu bionic-updates/main amd64 libc-ares2 amd64 1.14.0-1ubuntu0.1 [37.5 kB]\n",
            "Get:36 http://archive.ubuntu.com/ubuntu bionic-updates/universe amd64 nodejs amd64 8.10.0~dfsg-2ubuntu0.4 [4,796 kB]\n",
            "Fetched 9,743 kB in 0s (25.8 MB/s)\n",
            "Extracting templates from packages: 100%\n",
            "Selecting previously unselected package python-pkg-resources.\n",
            "(Reading database ... 155371 files and directories currently installed.)\n",
            "Preparing to unpack .../00-python-pkg-resources_39.0.1-2_all.deb ...\n",
            "Unpacking python-pkg-resources (39.0.1-2) ...\n",
            "Selecting previously unselected package python-six.\n",
            "Preparing to unpack .../01-python-six_1.11.0-2_all.deb ...\n",
            "Unpacking python-six (1.11.0-2) ...\n",
            "Selecting previously unselected package python-isodate.\n",
            "Preparing to unpack .../02-python-isodate_0.6.0-1_all.deb ...\n",
            "Unpacking python-isodate (0.6.0-1) ...\n",
            "Selecting previously unselected package python-pyparsing.\n",
            "Preparing to unpack .../03-python-pyparsing_2.2.0+dfsg1-2_all.deb ...\n",
            "Unpacking python-pyparsing (2.2.0+dfsg1-2) ...\n",
            "Selecting previously unselected package python-rdflib.\n",
            "Preparing to unpack .../04-python-rdflib_4.2.1-2_all.deb ...\n",
            "Unpacking python-rdflib (4.2.1-2) ...\n",
            "Selecting previously unselected package python-certifi.\n",
            "Preparing to unpack .../05-python-certifi_2018.1.18-2_all.deb ...\n",
            "Unpacking python-certifi (2018.1.18-2) ...\n",
            "Selecting previously unselected package python-chardet.\n",
            "Preparing to unpack .../06-python-chardet_3.0.4-1_all.deb ...\n",
            "Unpacking python-chardet (3.0.4-1) ...\n",
            "Selecting previously unselected package python-idna.\n",
            "Preparing to unpack .../07-python-idna_2.6-1_all.deb ...\n",
            "Unpacking python-idna (2.6-1) ...\n",
            "Selecting previously unselected package python-urllib3.\n",
            "Preparing to unpack .../08-python-urllib3_1.22-1ubuntu0.18.04.2_all.deb ...\n",
            "Unpacking python-urllib3 (1.22-1ubuntu0.18.04.2) ...\n",
            "Selecting previously unselected package python-requests.\n",
            "Preparing to unpack .../09-python-requests_2.18.4-2ubuntu0.1_all.deb ...\n",
            "Unpacking python-requests (2.18.4-2ubuntu0.1) ...\n",
            "Selecting previously unselected package python-ruamel.yaml.\n",
            "Preparing to unpack .../10-python-ruamel.yaml_0.15.34-1_amd64.deb ...\n",
            "Unpacking python-ruamel.yaml (0.15.34-1) ...\n",
            "Selecting previously unselected package python-cachecontrol.\n",
            "Preparing to unpack .../11-python-cachecontrol_0.11.7-1_all.deb ...\n",
            "Unpacking python-cachecontrol (0.11.7-1) ...\n",
            "Selecting previously unselected package python-lockfile.\n",
            "Preparing to unpack .../12-python-lockfile_1%3a0.12.2-2_all.deb ...\n",
            "Unpacking python-lockfile (1:0.12.2-2) ...\n",
            "Selecting previously unselected package python-mistune.\n",
            "Preparing to unpack .../13-python-mistune_0.8.3-2_all.deb ...\n",
            "Unpacking python-mistune (0.8.3-2) ...\n",
            "Selecting previously unselected package python-rdflib-jsonld.\n",
            "Preparing to unpack .../14-python-rdflib-jsonld_0.4.0-3_all.deb ...\n",
            "Unpacking python-rdflib-jsonld (0.4.0-3) ...\n",
            "Selecting previously unselected package python-typing.\n",
            "Preparing to unpack .../15-python-typing_3.6.2-1_all.deb ...\n",
            "Unpacking python-typing (3.6.2-1) ...\n",
            "Selecting previously unselected package python-avro.\n",
            "Preparing to unpack .../16-python-avro_1.8.2+dfsg-1_all.deb ...\n",
            "Unpacking python-avro (1.8.2+dfsg-1) ...\n",
            "Selecting previously unselected package libjs-bootstrap.\n",
            "Preparing to unpack .../17-libjs-bootstrap_3.3.7+dfsg-2_all.deb ...\n",
            "Unpacking libjs-bootstrap (3.3.7+dfsg-2) ...\n",
            "Selecting previously unselected package python-schema-salad.\n",
            "Preparing to unpack .../18-python-schema-salad_2.6.20171201034858-3_all.deb ...\n",
            "Unpacking python-schema-salad (2.6.20171201034858-3) ...\n",
            "Selecting previously unselected package python-shellescape.\n",
            "Preparing to unpack .../19-python-shellescape_3.4.1-2_all.deb ...\n",
            "Unpacking python-shellescape (3.4.1-2) ...\n",
            "Selecting previously unselected package cwltool.\n",
            "Preparing to unpack .../20-cwltool_1.0.20180302231433-1_all.deb ...\n",
            "Unpacking cwltool (1.0.20180302231433-1) ...\n",
            "Selecting previously unselected package libhts2:amd64.\n",
            "Preparing to unpack .../21-libhts2_1.7-2_amd64.deb ...\n",
            "Unpacking libhts2:amd64 (1.7-2) ...\n",
            "Selecting previously unselected package nodejs-doc.\n",
            "Preparing to unpack .../22-nodejs-doc_8.10.0~dfsg-2ubuntu0.4_all.deb ...\n",
            "Unpacking nodejs-doc (8.10.0~dfsg-2ubuntu0.4) ...\n",
            "Selecting previously unselected package python-asn1crypto.\n",
            "Preparing to unpack .../23-python-asn1crypto_0.24.0-1_all.deb ...\n",
            "Unpacking python-asn1crypto (0.24.0-1) ...\n",
            "Selecting previously unselected package python-cffi-backend.\n",
            "Preparing to unpack .../24-python-cffi-backend_1.11.5-1_amd64.deb ...\n",
            "Unpacking python-cffi-backend (1.11.5-1) ...\n",
            "Selecting previously unselected package python-enum34.\n",
            "Preparing to unpack .../25-python-enum34_1.1.6-2_all.deb ...\n",
            "Unpacking python-enum34 (1.1.6-2) ...\n",
            "Selecting previously unselected package python-ipaddress.\n",
            "Preparing to unpack .../26-python-ipaddress_1.0.17-1_all.deb ...\n",
            "Unpacking python-ipaddress (1.0.17-1) ...\n",
            "Selecting previously unselected package python-cryptography.\n",
            "Preparing to unpack .../27-python-cryptography_2.1.4-1ubuntu1.4_amd64.deb ...\n",
            "Unpacking python-cryptography (2.1.4-1ubuntu1.4) ...\n",
            "Selecting previously unselected package python-webencodings.\n",
            "Preparing to unpack .../28-python-webencodings_0.5-2_all.deb ...\n",
            "Unpacking python-webencodings (0.5-2) ...\n",
            "Selecting previously unselected package python-html5lib.\n",
            "Preparing to unpack .../29-python-html5lib_0.999999999-1_all.deb ...\n",
            "Unpacking python-html5lib (0.999999999-1) ...\n",
            "Selecting previously unselected package python-openssl.\n",
            "Preparing to unpack .../30-python-openssl_17.5.0-1ubuntu1_all.deb ...\n",
            "Unpacking python-openssl (17.5.0-1ubuntu1) ...\n",
            "Selecting previously unselected package python-sparqlwrapper.\n",
            "Preparing to unpack .../31-python-sparqlwrapper_1.7.6-2_all.deb ...\n",
            "Unpacking python-sparqlwrapper (1.7.6-2) ...\n",
            "Selecting previously unselected package samtools.\n",
            "Preparing to unpack .../32-samtools_1.7-1_amd64.deb ...\n",
            "Unpacking samtools (1.7-1) ...\n",
            "Selecting previously unselected package bowtie2.\n",
            "Preparing to unpack .../33-bowtie2_2.3.4.1-1_amd64.deb ...\n",
            "Unpacking bowtie2 (2.3.4.1-1) ...\n",
            "Selecting previously unselected package libc-ares2:amd64.\n",
            "Preparing to unpack .../34-libc-ares2_1.14.0-1ubuntu0.1_amd64.deb ...\n",
            "Unpacking libc-ares2:amd64 (1.14.0-1ubuntu0.1) ...\n",
            "Selecting previously unselected package nodejs.\n",
            "Preparing to unpack .../35-nodejs_8.10.0~dfsg-2ubuntu0.4_amd64.deb ...\n",
            "Unpacking nodejs (8.10.0~dfsg-2ubuntu0.4) ...\n",
            "Setting up python-idna (2.6-1) ...\n",
            "Setting up nodejs-doc (8.10.0~dfsg-2ubuntu0.4) ...\n",
            "Setting up python-asn1crypto (0.24.0-1) ...\n",
            "Setting up python-mistune (0.8.3-2) ...\n",
            "Setting up python-certifi (2018.1.18-2) ...\n",
            "Setting up python-avro (1.8.2+dfsg-1) ...\n",
            "Setting up python-pkg-resources (39.0.1-2) ...\n",
            "Setting up bowtie2 (2.3.4.1-1) ...\n",
            "Setting up python-ruamel.yaml (0.15.34-1) ...\n",
            "Setting up python-cffi-backend (1.11.5-1) ...\n",
            "Setting up python-six (1.11.0-2) ...\n",
            "Setting up python-isodate (0.6.0-1) ...\n",
            "Setting up python-shellescape (3.4.1-2) ...\n",
            "Setting up python-enum34 (1.1.6-2) ...\n",
            "Setting up libjs-bootstrap (3.3.7+dfsg-2) ...\n",
            "Setting up python-lockfile (1:0.12.2-2) ...\n",
            "Setting up python-ipaddress (1.0.17-1) ...\n",
            "Setting up libc-ares2:amd64 (1.14.0-1ubuntu0.1) ...\n",
            "Setting up libhts2:amd64 (1.7-2) ...\n",
            "Setting up python-webencodings (0.5-2) ...\n",
            "Setting up python-typing (3.6.2-1) ...\n",
            "Setting up python-pyparsing (2.2.0+dfsg1-2) ...\n",
            "Setting up python-urllib3 (1.22-1ubuntu0.18.04.2) ...\n",
            "Setting up python-chardet (3.0.4-1) ...\n",
            "Setting up nodejs (8.10.0~dfsg-2ubuntu0.4) ...\n",
            "update-alternatives: using /usr/bin/nodejs to provide /usr/bin/js (js) in auto mode\n",
            "Setting up python-rdflib (4.2.1-2) ...\n",
            "Setting up python-sparqlwrapper (1.7.6-2) ...\n",
            "Setting up python-rdflib-jsonld (0.4.0-3) ...\n",
            "Setting up python-cryptography (2.1.4-1ubuntu1.4) ...\n",
            "Setting up python-requests (2.18.4-2ubuntu0.1) ...\n",
            "Setting up python-cachecontrol (0.11.7-1) ...\n",
            "Setting up python-schema-salad (2.6.20171201034858-3) ...\n",
            "Setting up python-html5lib (0.999999999-1) ...\n",
            "Setting up samtools (1.7-1) ...\n",
            "Setting up python-openssl (17.5.0-1ubuntu1) ...\n",
            "Setting up cwltool (1.0.20180302231433-1) ...\n",
            "Processing triggers for libc-bin (2.27-3ubuntu1.3) ...\n",
            "/sbin/ldconfig.real: file /usr/local/lib/libjli.so is truncated\n",
            "\n",
            "/sbin/ldconfig.real: file /usr/local/lib/libnet.so is truncated\n",
            "\n",
            "/sbin/ldconfig.real: file /usr/local/lib/libjawt.so is truncated\n",
            "\n",
            "/sbin/ldconfig.real: /usr/local/lib/python3.7/dist-packages/ideep4py/lib/libmkldnn.so.0 is not a symbolic link\n",
            "\n",
            "Processing triggers for man-db (2.8.3-2ubuntu0.1) ...\n"
          ]
        }
      ],
      "source": [
        "!apt-get install bowtie2 samtools"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "JgTq4lFjq93N",
        "outputId": "36a0cf64-005f-40a7-b413-4b8b568bec7a"
      },
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Collecting package metadata (current_repodata.json): - \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\bdone\n",
            "Solving environment: - \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\bdone\n",
            "\n",
            "## Package Plan ##\n",
            "\n",
            "  environment location: /usr/local\n",
            "\n",
            "  added / updated specs:\n",
            "    - bowtie2\n",
            "\n",
            "\n",
            "The following packages will be downloaded:\n",
            "\n",
            "    package                    |            build\n",
            "    ---------------------------|-----------------\n",
            "    bowtie2-2.3.5.1            |   py37he513fc3_0        12.0 MB  bioconda\n",
            "    perl-5.32.1                | 0_h7f98852_perl5        14.5 MB  conda-forge\n",
            "    tbb-2020.2                 |       h4bd325d_4         1.5 MB  conda-forge\n",
            "    ------------------------------------------------------------\n",
            "                                           Total:        27.9 MB\n",
            "\n",
            "The following NEW packages will be INSTALLED:\n",
            "\n",
            "  bowtie2            bioconda/linux-64::bowtie2-2.3.5.1-py37he513fc3_0\n",
            "  perl               conda-forge/linux-64::perl-5.32.1-0_h7f98852_perl5\n",
            "  tbb                conda-forge/linux-64::tbb-2020.2-h4bd325d_4\n",
            "\n",
            "\n",
            "\n",
            "Downloading and Extracting Packages\n",
            "perl-5.32.1          | 14.5 MB   | : 100% 1.0/1 [00:02<00:00,  2.81s/it]\n",
            "bowtie2-2.3.5.1      | 12.0 MB   | : 100% 1.0/1 [00:01<00:00,  1.78s/it]\n",
            "tbb-2020.2           | 1.5 MB    | : 100% 1.0/1 [00:00<00:00,  4.11it/s]\n",
            "Preparing transaction: / \b\bdone\n",
            "Verifying transaction: \\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\bdone\n",
            "Executing transaction: | \b\b/ \b\bdone\n"
          ]
        }
      ],
      "source": [
        "!conda install -c bioconda bowtie2"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "vcr5CjibHTgF"
      },
      "source": [
        "## Установка FastQC"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "62l1Q2ymbENA",
        "outputId": "fb0816c3-198b-4c0b-ebcd-feb084b50a83"
      },
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "openjdk version \"1.8.0_312\"\n",
            "OpenJDK Runtime Environment (Zulu 8.58.0.13-CA-linux64) (build 1.8.0_312-b07)\n",
            "OpenJDK 64-Bit Server VM (Zulu 8.58.0.13-CA-linux64) (build 25.312-b07, mixed mode)\n"
          ]
        }
      ],
      "source": [
        "!java -version"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "8bpgwFgcbgRd",
        "outputId": "9a1781b8-9d67-4de4-841d-6c5e5e9eadec"
      },
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "--2022-02-28 17:26:26--  https://www.bioinformatics.babraham.ac.uk/projects/fastqc/fastqc_v0.11.9.zip\n",
            "Resolving www.bioinformatics.babraham.ac.uk (www.bioinformatics.babraham.ac.uk)... 149.155.133.4\n",
            "Connecting to www.bioinformatics.babraham.ac.uk (www.bioinformatics.babraham.ac.uk)|149.155.133.4|:443... connected.\n",
            "HTTP request sent, awaiting response... 200 OK\n",
            "Length: 10249221 (9.8M) [application/zip]\n",
            "Saving to: ‘fastqc_v0.11.9.zip’\n",
            "\n",
            "fastqc_v0.11.9.zip  100%[===================>]   9.77M  6.94MB/s    in 1.4s    \n",
            "\n",
            "2022-02-28 17:26:27 (6.94 MB/s) - ‘fastqc_v0.11.9.zip’ saved [10249221/10249221]\n",
            "\n"
          ]
        }
      ],
      "source": [
        "!wget https://www.bioinformatics.babraham.ac.uk/projects/fastqc/fastqc_v0.11.9.zip"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "yTcTjOTJblcm",
        "outputId": "34f95263-6660-460d-af60-5d986b1e9a87"
      },
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Archive:  fastqc_v0.11.9.zip\n",
            "  inflating: FastQC/cisd-jhdf5.jar   \n",
            "   creating: FastQC/Configuration/\n",
            "  inflating: FastQC/Configuration/adapter_list.txt  \n",
            "  inflating: FastQC/Configuration/contaminant_list.txt  \n",
            "  inflating: FastQC/Configuration/limits.txt  \n",
            "  inflating: FastQC/fastqc           \n",
            "  inflating: FastQC/fastqc_icon.ico  \n",
            "   creating: FastQC/Help/\n",
            "   creating: FastQC/Help/1 Introduction/\n",
            "   creating: FastQC/Help/1 Introduction/.svn/\n",
            "  inflating: FastQC/Help/1 Introduction/.svn/entries  \n",
            "   creating: FastQC/Help/1 Introduction/.svn/props/\n",
            "   creating: FastQC/Help/1 Introduction/.svn/text-base/\n",
            "  inflating: FastQC/Help/1 Introduction/.svn/text-base/1.1 What is FastQC.html.svn-base  \n",
            "   creating: FastQC/Help/1 Introduction/.svn/tmp/\n",
            "   creating: FastQC/Help/1 Introduction/.svn/tmp/props/\n",
            "  inflating: FastQC/Help/1 Introduction/1.1 What is FastQC.html  \n",
            "   creating: FastQC/Help/2 Basic Operations/\n",
            "   creating: FastQC/Help/2 Basic Operations/.svn/\n",
            "  inflating: FastQC/Help/2 Basic Operations/.svn/entries  \n",
            "   creating: FastQC/Help/2 Basic Operations/.svn/props/\n",
            "   creating: FastQC/Help/2 Basic Operations/.svn/text-base/\n",
            "  inflating: FastQC/Help/2 Basic Operations/.svn/text-base/2.1 Opening a sequence file.html.svn-base  \n",
            "  inflating: FastQC/Help/2 Basic Operations/.svn/text-base/2.2 Evaluating Results.html.svn-base  \n",
            "  inflating: FastQC/Help/2 Basic Operations/.svn/text-base/2.3 Saving a Report.html.svn-base  \n",
            "   creating: FastQC/Help/2 Basic Operations/.svn/tmp/\n",
            "   creating: FastQC/Help/2 Basic Operations/.svn/tmp/props/\n",
            "  inflating: FastQC/Help/2 Basic Operations/2.1 Opening a sequence file.html  \n",
            "  inflating: FastQC/Help/2 Basic Operations/2.2 Evaluating Results.html  \n",
            "  inflating: FastQC/Help/2 Basic Operations/2.3 Saving a Report.html  \n",
            "   creating: FastQC/Help/3 Analysis Modules/\n",
            "   creating: FastQC/Help/3 Analysis Modules/.svn/\n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/entries  \n",
            "   creating: FastQC/Help/3 Analysis Modules/.svn/prop-base/\n",
            " extracting: FastQC/Help/3 Analysis Modules/.svn/prop-base/duplication_levels.png.svn-base  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/prop-base/kmer_profiles.png.svn-base  \n",
            " extracting: FastQC/Help/3 Analysis Modules/.svn/prop-base/per_base_gc_content.png.svn-base  \n",
            " extracting: FastQC/Help/3 Analysis Modules/.svn/prop-base/per_base_n_content.png.svn-base  \n",
            " extracting: FastQC/Help/3 Analysis Modules/.svn/prop-base/per_base_quality.png.svn-base  \n",
            " extracting: FastQC/Help/3 Analysis Modules/.svn/prop-base/per_base_sequence_content.png.svn-base  \n",
            " extracting: FastQC/Help/3 Analysis Modules/.svn/prop-base/per_sequence_gc_content.png.svn-base  \n",
            " extracting: FastQC/Help/3 Analysis Modules/.svn/prop-base/per_sequence_quality.png.svn-base  \n",
            " extracting: FastQC/Help/3 Analysis Modules/.svn/prop-base/per_tile_quality.png.svn-base  \n",
            " extracting: FastQC/Help/3 Analysis Modules/.svn/prop-base/sequence_length_distribution.png.svn-base  \n",
            "   creating: FastQC/Help/3 Analysis Modules/.svn/props/\n",
            "   creating: FastQC/Help/3 Analysis Modules/.svn/text-base/\n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/1 Basic Statistics.html.svn-base  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/10 Adapter Content.html.svn-base  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/11 Kmer Content.html.svn-base  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/12 Per Tile Sequence Quality.html.svn-base  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/2 Per Base Sequence Quality.html.svn-base  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/3 Per Sequence Quality Scores.html.svn-base  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/4 Per Base Sequence Content.html.svn-base  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/5 Per Sequence GC Content.html.svn-base  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/6 Per Base N Content.html.svn-base  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/7 Sequence Length Distribution.html.svn-base  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/8 Duplicate Sequences.html.svn-base  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/9 Overrepresented Sequences.html.svn-base  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/duplication_levels.png.svn-base  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/kmer_profiles.png.svn-base  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/per_base_gc_content.png.svn-base  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/per_base_n_content.png.svn-base  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/per_base_quality.png.svn-base  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/per_base_sequence_content.png.svn-base  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/per_sequence_gc_content.png.svn-base  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/per_sequence_quality.png.svn-base  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/per_tile_quality.png.svn-base  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/sequence_length_distribution.png.svn-base  \n",
            "   creating: FastQC/Help/3 Analysis Modules/.svn/tmp/\n",
            "   creating: FastQC/Help/3 Analysis Modules/.svn/tmp/props/\n",
            "  inflating: FastQC/Help/3 Analysis Modules/1 Basic Statistics.html  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/10 Adapter Content.html  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/11 Kmer Content.html  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/12 Per Tile Sequence Quality.html  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/2 Per Base Sequence Quality.html  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/3 Per Sequence Quality Scores.html  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/4 Per Base Sequence Content.html  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/5 Per Sequence GC Content.html  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/6 Per Base N Content.html  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/7 Sequence Length Distribution.html  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/8 Duplicate Sequences.html  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/9 Overrepresented Sequences.html  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/duplication_levels.png  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/kmer_profiles.png  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/per_base_gc_content.png  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/per_base_n_content.png  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/per_base_quality.png  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/per_base_sequence_content.png  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/per_sequence_gc_content.png  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/per_sequence_quality.png  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/per_tile_quality.png  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/sequence_length_distribution.png  \n",
            "  inflating: FastQC/INSTALL.txt      \n",
            "  inflating: FastQC/jbzip2-0.9.jar   \n",
            "  inflating: FastQC/LICENSE          \n",
            "  inflating: FastQC/LICENSE.txt      \n",
            "  inflating: FastQC/LICENSE_JHDF5.txt  \n",
            "   creating: FastQC/net/\n",
            "   creating: FastQC/net/sourceforge/\n",
            "   creating: FastQC/net/sourceforge/iharder/\n",
            "   creating: FastQC/net/sourceforge/iharder/base64/\n",
            "  inflating: FastQC/net/sourceforge/iharder/base64/Base64$1.class  \n",
            "  inflating: FastQC/net/sourceforge/iharder/base64/Base64$InputStream.class  \n",
            "  inflating: FastQC/net/sourceforge/iharder/base64/Base64$OutputStream.class  \n",
            "  inflating: FastQC/net/sourceforge/iharder/base64/Base64.class  \n",
            "   creating: FastQC/org/\n",
            "   creating: FastQC/org/apache/\n",
            "   creating: FastQC/org/apache/commons/\n",
            "   creating: FastQC/org/apache/commons/math3/\n",
            "   creating: FastQC/org/apache/commons/math3/analysis/\n",
            "   creating: FastQC/org/apache/commons/math3/analysis/solvers/\n",
            "  inflating: FastQC/org/apache/commons/math3/analysis/solvers/AbstractUnivariateSolver.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/analysis/solvers/AllowedSolution.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/analysis/solvers/BaseAbstractUnivariateSolver.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/analysis/solvers/BaseUnivariateSolver.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/analysis/solvers/BracketedUnivariateSolver.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/analysis/solvers/BrentSolver.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/analysis/solvers/UnivariateSolver.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/analysis/solvers/UnivariateSolverUtils.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/analysis/UnivariateFunction.class  \n",
            "   creating: FastQC/org/apache/commons/math3/distribution/\n",
            "  inflating: FastQC/org/apache/commons/math3/distribution/AbstractIntegerDistribution.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/distribution/AbstractRealDistribution$1.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/distribution/AbstractRealDistribution.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/distribution/BetaDistribution.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/distribution/BinomialDistribution.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/distribution/CauchyDistribution.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/distribution/ChiSquaredDistribution.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/distribution/FDistribution.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/distribution/GammaDistribution.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/distribution/HypergeometricDistribution.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/distribution/IntegerDistribution.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/distribution/NormalDistribution.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/distribution/PascalDistribution.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/distribution/PoissonDistribution.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/distribution/RealDistribution.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/distribution/SaddlePointExpansion.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/distribution/TDistribution.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/distribution/WeibullDistribution.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/distribution/ZipfDistribution.class  \n",
            "   creating: FastQC/org/apache/commons/math3/exception/\n",
            "  inflating: FastQC/org/apache/commons/math3/exception/ConvergenceException.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/exception/DimensionMismatchException.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/exception/MathArithmeticException.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/exception/MathIllegalArgumentException.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/exception/MathIllegalNumberException.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/exception/MathIllegalStateException.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/exception/MathInternalError.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/exception/MaxCountExceededException.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/exception/NoBracketingException.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/exception/NotFiniteNumberException.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/exception/NotPositiveException.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/exception/NotStrictlyPositiveException.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/exception/NullArgumentException.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/exception/NumberIsTooLargeException.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/exception/NumberIsTooSmallException.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/exception/OutOfRangeException.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/exception/TooManyEvaluationsException.class  \n",
            "   creating: FastQC/org/apache/commons/math3/exception/util/\n",
            "  inflating: FastQC/org/apache/commons/math3/exception/util/ArgUtils.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/exception/util/ExceptionContext.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/exception/util/ExceptionContextProvider.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/exception/util/Localizable.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/exception/util/LocalizedFormats.class  \n",
            "   creating: FastQC/org/apache/commons/math3/random/\n",
            "  inflating: FastQC/org/apache/commons/math3/random/AbstractWell.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/random/BitsStreamGenerator.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/random/RandomData.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/random/RandomDataImpl.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/random/RandomGenerator.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/random/Well19937c.class  \n",
            "   creating: FastQC/org/apache/commons/math3/special/\n",
            "  inflating: FastQC/org/apache/commons/math3/special/Beta$1.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/special/Beta.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/special/Erf.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/special/Gamma$1.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/special/Gamma.class  \n",
            "   creating: FastQC/org/apache/commons/math3/util/\n",
            "  inflating: FastQC/org/apache/commons/math3/util/ArithmeticUtils.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/util/ContinuedFraction.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/util/DoubleArray.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/util/FastMath$ExpFracTable.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/util/FastMath$ExpIntTable.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/util/FastMath$lnMant.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/util/FastMath.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/util/FastMathCalc.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/util/FastMathLiteralArrays.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/util/Incrementor$1.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/util/Incrementor$MaxCountExceededCallback.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/util/Incrementor.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/util/MathUtils.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/util/Precision.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/util/ResizableDoubleArray.class  \n",
            "  inflating: FastQC/README.md        \n",
            "  inflating: FastQC/README.txt       \n",
            "  inflating: FastQC/RELEASE_NOTES.txt  \n",
            "  inflating: FastQC/run_fastqc.bat   \n",
            "  inflating: FastQC/sam-1.103.jar    \n",
            "   creating: FastQC/Templates/\n",
            "  inflating: FastQC/Templates/fastqc2fo.xsl  \n",
            "  inflating: FastQC/Templates/header_template.html  \n",
            "   creating: FastQC/Templates/Icons/\n",
            " extracting: FastQC/Templates/Icons/error.png  \n",
            " extracting: FastQC/Templates/Icons/fastqc_icon.png  \n",
            " extracting: FastQC/Templates/Icons/tick.png  \n",
            " extracting: FastQC/Templates/Icons/warning.png  \n",
            "   creating: FastQC/uk/\n",
            "   creating: FastQC/uk/ac/\n",
            "   creating: FastQC/uk/ac/babraham/\n",
            "   creating: FastQC/uk/ac/babraham/FastQC/\n",
            "   creating: FastQC/uk/ac/babraham/FastQC/Analysis/\n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Analysis/AnalysisListener.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Analysis/AnalysisQueue.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Analysis/AnalysisRunner.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Analysis/OfflineRunner.class  \n",
            "   creating: FastQC/uk/ac/babraham/FastQC/Dialogs/\n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Dialogs/AboutDialog$1.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Dialogs/AboutDialog.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Dialogs/FastQCTitlePanel$SmoothJLabel.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Dialogs/FastQCTitlePanel.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Dialogs/WelcomePanel.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/FastQCApplication$1.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/FastQCApplication.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/FastQCConfig.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/FastQCMenuBar.class  \n",
            "   creating: FastQC/uk/ac/babraham/FastQC/FileFilters/\n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/FileFilters/BAMFileFilter.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/FileFilters/CasavaFastQFileFilter.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/FileFilters/FastQFileFilter.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/FileFilters/GobyFileFilter.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/FileFilters/MappedBAMFileFilter.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/FileFilters/SequenceFileFilter.class  \n",
            "   creating: FastQC/uk/ac/babraham/FastQC/Graphs/\n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Graphs/BaseGroup.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Graphs/LineGraph.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Graphs/QualityBoxPlot.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Graphs/TileGraph.class  \n",
            "   creating: FastQC/uk/ac/babraham/FastQC/Help/\n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Help/HelpDialog.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Help/HelpIndexRoot$FileSorter.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Help/HelpIndexRoot.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Help/HelpPage.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Help/HelpPageDisplay$HelpEditor.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Help/HelpPageDisplay.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Help/HelpSearchPanel.class  \n",
            "   creating: FastQC/uk/ac/babraham/FastQC/Modules/\n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/AbstractQCModule.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/AdapterContent$Adapter.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/AdapterContent$ResultsTable.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/AdapterContent.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/BasicStats$ResultsTable.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/BasicStats.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/DuplicationLevel.class  \n",
            "   creating: FastQC/uk/ac/babraham/FastQC/Modules/GCModel/\n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/GCModel/GCModel.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/GCModel/GCModelValue.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/KmerContent$Kmer.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/KmerContent$ResultsTable.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/KmerContent.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/ModuleConfig.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/ModuleFactory.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/NContent.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/OverRepresentedSeqs$OverrepresentedSeq.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/OverRepresentedSeqs$ResultsTable.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/OverRepresentedSeqs.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/PerBaseQualityScores.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/PerBaseSequenceContent.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/PerSequenceGCContent.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/PerSequenceQualityScores.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/PerTileQualityScores.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/QCModule.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/SequenceLengthDistribution.class  \n",
            "   creating: FastQC/uk/ac/babraham/FastQC/Report/\n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Report/HTMLReportArchive.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Report/stylesheet.txt  \n",
            "   creating: FastQC/uk/ac/babraham/FastQC/Resources/\n",
            " extracting: FastQC/uk/ac/babraham/FastQC/Resources/error.png  \n",
            " extracting: FastQC/uk/ac/babraham/FastQC/Resources/fastqc_icon.png  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Resources/fastqc_icon.svg  \n",
            " extracting: FastQC/uk/ac/babraham/FastQC/Resources/fastqc_icon_100.png  \n",
            " extracting: FastQC/uk/ac/babraham/FastQC/Resources/tick.png  \n",
            " extracting: FastQC/uk/ac/babraham/FastQC/Resources/warning.png  \n",
            "   creating: FastQC/uk/ac/babraham/FastQC/Results/\n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Results/ResultsPanel$ModuleRenderer.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Results/ResultsPanel.class  \n",
            "   creating: FastQC/uk/ac/babraham/FastQC/Sequence/\n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Sequence/BAMFile.class  \n",
            "   creating: FastQC/uk/ac/babraham/FastQC/Sequence/Contaminant/\n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Sequence/Contaminant/Contaminant.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Sequence/Contaminant/ContaminantHit.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Sequence/Contaminant/ContaminentFinder.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Sequence/Fast5File.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Sequence/FastQFile.class  \n",
            "   creating: FastQC/uk/ac/babraham/FastQC/Sequence/QualityEncoding/\n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Sequence/QualityEncoding/PhredEncoding.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Sequence/Sequence.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Sequence/SequenceFactory.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Sequence/SequenceFile.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Sequence/SequenceFileGroup.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Sequence/SequenceFormatException.class  \n",
            "   creating: FastQC/uk/ac/babraham/FastQC/Statistics/\n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Statistics/NormalDistribution.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Statistics/PearsonCorrelation.class  \n",
            "   creating: FastQC/uk/ac/babraham/FastQC/Utilities/\n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Utilities/CasavaBasename.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Utilities/HotColdColourGradient.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Utilities/ImageToBase64.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Utilities/MultiMemberGZIPInputStream.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Utilities/NameFormatException.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Utilities/NanoporeBasename.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Utilities/QualityCount.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Utilities/RGB.class  \n"
          ]
        }
      ],
      "source": [
        "!unzip fastqc_v0.11.9.zip"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "id": "S79LH0aXbvLx"
      },
      "outputs": [],
      "source": [
        "!chmod a+x FastQC/fastqc"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "PsLuZDt4b7jT",
        "outputId": "0b19e803-8e3e-4125-da4e-12cd25220530"
      },
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "\n",
            "            FastQC - A high throughput sequence QC analysis tool\n",
            "\n",
            "SYNOPSIS\n",
            "\n",
            "\tfastqc seqfile1 seqfile2 .. seqfileN\n",
            "\n",
            "    fastqc [-o output dir] [--(no)extract] [-f fastq|bam|sam] \n",
            "           [-c contaminant file] seqfile1 .. seqfileN\n",
            "\n",
            "DESCRIPTION\n",
            "\n",
            "    FastQC reads a set of sequence files and produces from each one a quality\n",
            "    control report consisting of a number of different modules, each one of \n",
            "    which will help to identify a different potential type of problem in your\n",
            "    data.\n",
            "    \n",
            "    If no files to process are specified on the command line then the program\n",
            "    will start as an interactive graphical application.  If files are provided\n",
            "    on the command line then the program will run with no user interaction\n",
            "    required.  In this mode it is suitable for inclusion into a standardised\n",
            "    analysis pipeline.\n",
            "    \n",
            "    The options for the program as as follows:\n",
            "    \n",
            "    -h --help       Print this help file and exit\n",
            "    \n",
            "    -v --version    Print the version of the program and exit\n",
            "    \n",
            "    -o --outdir     Create all output files in the specified output directory.\n",
            "                    Please note that this directory must exist as the program\n",
            "                    will not create it.  If this option is not set then the \n",
            "                    output file for each sequence file is created in the same\n",
            "                    directory as the sequence file which was processed.\n",
            "                    \n",
            "    --casava        Files come from raw casava output. Files in the same sample\n",
            "                    group (differing only by the group number) will be analysed\n",
            "                    as a set rather than individually. Sequences with the filter\n",
            "                    flag set in the header will be excluded from the analysis.\n",
            "                    Files must have the same names given to them by casava\n",
            "                    (including being gzipped and ending with .gz) otherwise they\n",
            "                    won't be grouped together correctly.\n",
            "                    \n",
            "    --nano          Files come from nanopore sequences and are in fast5 format. In\n",
            "                    this mode you can pass in directories to process and the program\n",
            "                    will take in all fast5 files within those directories and produce\n",
            "                    a single output file from the sequences found in all files.                    \n",
            "                    \n",
            "    --nofilter      If running with --casava then don't remove read flagged by\n",
            "                    casava as poor quality when performing the QC analysis.\n",
            "                   \n",
            "    --extract       If set then the zipped output file will be uncompressed in\n",
            "                    the same directory after it has been created.  By default\n",
            "                    this option will be set if fastqc is run in non-interactive\n",
            "                    mode.\n",
            "                    \n",
            "    -j --java       Provides the full path to the java binary you want to use to\n",
            "                    launch fastqc. If not supplied then java is assumed to be in\n",
            "                    your path.\n",
            "                   \n",
            "    --noextract     Do not uncompress the output file after creating it.  You\n",
            "                    should set this option if you do not wish to uncompress\n",
            "                    the output when running in non-interactive mode.\n",
            "                    \n",
            "    --nogroup       Disable grouping of bases for reads >50bp. All reports will\n",
            "                    show data for every base in the read.  WARNING: Using this\n",
            "                    option will cause fastqc to crash and burn if you use it on\n",
            "                    really long reads, and your plots may end up a ridiculous size.\n",
            "                    You have been warned!\n",
            "                    \n",
            "    --min_length    Sets an artificial lower limit on the length of the sequence\n",
            "                    to be shown in the report.  As long as you set this to a value\n",
            "                    greater or equal to your longest read length then this will be\n",
            "                    the sequence length used to create your read groups.  This can\n",
            "                    be useful for making directly comaparable statistics from \n",
            "                    datasets with somewhat variable read lengths.\n",
            "                    \n",
            "    -f --format     Bypasses the normal sequence file format detection and\n",
            "                    forces the program to use the specified format.  Valid\n",
            "                    formats are bam,sam,bam_mapped,sam_mapped and fastq\n",
            "                    \n",
            "    -t --threads    Specifies the number of files which can be processed\n",
            "                    simultaneously.  Each thread will be allocated 250MB of\n",
            "                    memory so you shouldn't run more threads than your\n",
            "                    available memory will cope with, and not more than\n",
            "                    6 threads on a 32 bit machine\n",
            "                  \n",
            "    -c              Specifies a non-default file which contains the list of\n",
            "    --contaminants  contaminants to screen overrepresented sequences against.\n",
            "                    The file must contain sets of named contaminants in the\n",
            "                    form name[tab]sequence.  Lines prefixed with a hash will\n",
            "                    be ignored.\n",
            "\n",
            "    -a              Specifies a non-default file which contains the list of\n",
            "    --adapters      adapter sequences which will be explicity searched against\n",
            "                    the library. The file must contain sets of named adapters\n",
            "                    in the form name[tab]sequence.  Lines prefixed with a hash\n",
            "                    will be ignored.\n",
            "                    \n",
            "    -l              Specifies a non-default file which contains a set of criteria\n",
            "    --limits        which will be used to determine the warn/error limits for the\n",
            "                    various modules.  This file can also be used to selectively \n",
            "                    remove some modules from the output all together.  The format\n",
            "                    needs to mirror the default limits.txt file found in the\n",
            "                    Configuration folder.\n",
            "                    \n",
            "   -k --kmers       Specifies the length of Kmer to look for in the Kmer content\n",
            "                    module. Specified Kmer length must be between 2 and 10. Default\n",
            "                    length is 7 if not specified.\n",
            "                    \n",
            "   -q --quiet       Supress all progress messages on stdout and only report errors.\n",
            "   \n",
            "   -d --dir         Selects a directory to be used for temporary files written when\n",
            "                    generating report images. Defaults to system temp directory if\n",
            "                    not specified.\n",
            "                    \n",
            "BUGS\n",
            "\n",
            "    Any bugs in fastqc should be reported either to simon.andrews@babraham.ac.uk\n",
            "    or in www.bioinformatics.babraham.ac.uk/bugzilla/\n",
            "                   \n",
            "    \n"
          ]
        }
      ],
      "source": [
        "!./FastQC/fastqc --help"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "10Fs3qrmHmJi"
      },
      "source": [
        "# Выравнивание"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "-z4GQiMfU1wf"
      },
      "source": [
        "## Скачиваем чтения\n"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "lFWnMg0uU0sJ",
        "outputId": "3cc5158f-a2dc-4d65-ed05-c5f5f20c3933"
      },
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "--2022-02-28 18:06:00--  https://www.encodeproject.org/files/ENCFF052DPD/@@download/ENCFF052DPD.fastq.gz\n",
            "Resolving www.encodeproject.org (www.encodeproject.org)... 34.211.244.144\n",
            "Connecting to www.encodeproject.org (www.encodeproject.org)|34.211.244.144|:443... connected.\n",
            "HTTP request sent, awaiting response... 307 Temporary Redirect\n",
            "Location: https://encode-public.s3.amazonaws.com/2016/05/30/b228675c-53fa-426e-aec7-0b9b12ad9d90/ENCFF052DPD.fastq.gz?response-content-disposition=attachment%3B%20filename%3DENCFF052DPD.fastq.gz&AWSAccessKeyId=ASIATGZNGCNXZ432VCC3&Signature=Aj9X1iIK4UfsH8vwouuFTXGfoig%3D&x-amz-security-token=IQoJb3JpZ2luX2VjEJn%2F%2F%2F%2F%2F%2F%2F%2F%2F%2FwEaCXVzLXdlc3QtMiJHMEUCIQDmnJF9590s83BJbzBdCqsc3RBzn2enFgxNa4CnPNEJ3QIgD15O8os5NzmgeTmlTFbBi4vbTeFEsjwgF9BkiIUgeR0qgwQI8v%2F%2F%2F%2F%2F%2F%2F%2F%2F%2FARAAGgwyMjA3NDg3MTQ4NjMiDN8a%2B5gze9YE%2Ft%2BMrSrXAzhAe4GZoAQW7VNYDegEgTrvoUpm%2F9LoX0GLQPp%2Bu3SsceQJhegdpJXvK4G5XV1wKBSYmdXT0lNA%2FJCszMuYezMtdeO5yoIQmzJZIO1%2FU7WGTtBcMujHuLkSMwOyafnCqGP11fWLwPtGx1X8nDSln8F3Q28DqmRxt0g%2FXMk35jTLSIAFBX7k%2FrNk2UhN0xx%2Fuy4DIG8cvr0dKUDE0pA7spuHkS8tnCgX4QU8byhYkGCXx1sOkNCjNh0nVJbv8WAJTVs4HPo%2FEcQkaeftYVgQ%2FjodjHFJDvkMFfbjHE8666IwFlt4Zqr5YJ%2FQ5z8DCl1UbV7sJ8vIvAcYdzUI46JYWvhmRFOo4M35JfYNJRrs4wizUb0krl7DA4WTRdJ0EZXQ1y8%2FRtrYTeysE22UUUw%2BK3wcrZULRXCmjKWHskE%2FVg6xEdTXufTRyAWwcvoGV13LfdNh6jzL5J6J1E5qkKiq%2BMjM1W9fZi%2FL9dv8xwcyw%2FH6TmmBeUyskD8BjrLUkRNDRY9%2B4LYWoNIZqSj69E39CIrfhT5%2B1Kq%2Betp3f5%2FG0NWXDgFx5YVzgkP8ng%2FCRefBL%2F9I%2BbYR5X0%2B9JljJ6L21401%2FELZzBqCOkcaHisszBnOb7tCSA96rjCxhPSQBjqlAQWRXFnzpgBYc1JKv0PQzR3gJqobM965KFlESm7EORYSDeYQAJJeYU3%2BMcaXBNcwuKjQS0PGBrXURXV8QXfCDp7YuEOis5LV6mQRv5YRASu6gbT1nQsgzV1diEkVh%2FXKG6QyzqWL3P1waeTrpQoO8lmfU%2F15ryvHmxGxR8AQmG0TW5up9QWFo0zSkJsGhxjDvclITIIFuatwc9ua9uCzJein%2FpJpXA%3D%3D&Expires=1646201161 [following]\n",
            "--2022-02-28 18:06:01--  https://encode-public.s3.amazonaws.com/2016/05/30/b228675c-53fa-426e-aec7-0b9b12ad9d90/ENCFF052DPD.fastq.gz?response-content-disposition=attachment%3B%20filename%3DENCFF052DPD.fastq.gz&AWSAccessKeyId=ASIATGZNGCNXZ432VCC3&Signature=Aj9X1iIK4UfsH8vwouuFTXGfoig%3D&x-amz-security-token=IQoJb3JpZ2luX2VjEJn%2F%2F%2F%2F%2F%2F%2F%2F%2F%2FwEaCXVzLXdlc3QtMiJHMEUCIQDmnJF9590s83BJbzBdCqsc3RBzn2enFgxNa4CnPNEJ3QIgD15O8os5NzmgeTmlTFbBi4vbTeFEsjwgF9BkiIUgeR0qgwQI8v%2F%2F%2F%2F%2F%2F%2F%2F%2F%2FARAAGgwyMjA3NDg3MTQ4NjMiDN8a%2B5gze9YE%2Ft%2BMrSrXAzhAe4GZoAQW7VNYDegEgTrvoUpm%2F9LoX0GLQPp%2Bu3SsceQJhegdpJXvK4G5XV1wKBSYmdXT0lNA%2FJCszMuYezMtdeO5yoIQmzJZIO1%2FU7WGTtBcMujHuLkSMwOyafnCqGP11fWLwPtGx1X8nDSln8F3Q28DqmRxt0g%2FXMk35jTLSIAFBX7k%2FrNk2UhN0xx%2Fuy4DIG8cvr0dKUDE0pA7spuHkS8tnCgX4QU8byhYkGCXx1sOkNCjNh0nVJbv8WAJTVs4HPo%2FEcQkaeftYVgQ%2FjodjHFJDvkMFfbjHE8666IwFlt4Zqr5YJ%2FQ5z8DCl1UbV7sJ8vIvAcYdzUI46JYWvhmRFOo4M35JfYNJRrs4wizUb0krl7DA4WTRdJ0EZXQ1y8%2FRtrYTeysE22UUUw%2BK3wcrZULRXCmjKWHskE%2FVg6xEdTXufTRyAWwcvoGV13LfdNh6jzL5J6J1E5qkKiq%2BMjM1W9fZi%2FL9dv8xwcyw%2FH6TmmBeUyskD8BjrLUkRNDRY9%2B4LYWoNIZqSj69E39CIrfhT5%2B1Kq%2Betp3f5%2FG0NWXDgFx5YVzgkP8ng%2FCRefBL%2F9I%2BbYR5X0%2B9JljJ6L21401%2FELZzBqCOkcaHisszBnOb7tCSA96rjCxhPSQBjqlAQWRXFnzpgBYc1JKv0PQzR3gJqobM965KFlESm7EORYSDeYQAJJeYU3%2BMcaXBNcwuKjQS0PGBrXURXV8QXfCDp7YuEOis5LV6mQRv5YRASu6gbT1nQsgzV1diEkVh%2FXKG6QyzqWL3P1waeTrpQoO8lmfU%2F15ryvHmxGxR8AQmG0TW5up9QWFo0zSkJsGhxjDvclITIIFuatwc9ua9uCzJein%2FpJpXA%3D%3D&Expires=1646201161\n",
            "Resolving encode-public.s3.amazonaws.com (encode-public.s3.amazonaws.com)... 52.218.136.67\n",
            "Connecting to encode-public.s3.amazonaws.com (encode-public.s3.amazonaws.com)|52.218.136.67|:443... connected.\n",
            "HTTP request sent, awaiting response... 200 OK\n",
            "Length: 894965699 (854M) [binary/octet-stream]\n",
            "Saving to: ‘ENCFF052DPD.fastq.gz.1’\n",
            "\n",
            "ENCFF052DPD.fastq.g 100%[===================>] 853.50M  16.8MB/s    in 73s     \n",
            "\n",
            "2022-02-28 18:07:15 (11.7 MB/s) - ‘ENCFF052DPD.fastq.gz.1’ saved [894965699/894965699]\n",
            "\n",
            "--2022-02-28 18:07:15--  https://www.encodeproject.org/files/ENCFF726ZWD/@@download/ENCFF726ZWD.fastq.gz\n",
            "Resolving www.encodeproject.org (www.encodeproject.org)... 34.211.244.144\n",
            "Connecting to www.encodeproject.org (www.encodeproject.org)|34.211.244.144|:443... connected.\n",
            "HTTP request sent, awaiting response... 307 Temporary Redirect\n",
            "Location: https://encode-public.s3.amazonaws.com/2016/05/30/81e75b9d-f571-43ef-b644-2cf4dfb2ddbc/ENCFF726ZWD.fastq.gz?response-content-disposition=attachment%3B%20filename%3DENCFF726ZWD.fastq.gz&AWSAccessKeyId=ASIATGZNGCNXZ432VCC3&Signature=QR821mJ9xHPZvn8Uk%2Fo2cbRiVXk%3D&x-amz-security-token=IQoJb3JpZ2luX2VjEJn%2F%2F%2F%2F%2F%2F%2F%2F%2F%2FwEaCXVzLXdlc3QtMiJHMEUCIQDmnJF9590s83BJbzBdCqsc3RBzn2enFgxNa4CnPNEJ3QIgD15O8os5NzmgeTmlTFbBi4vbTeFEsjwgF9BkiIUgeR0qgwQI8v%2F%2F%2F%2F%2F%2F%2F%2F%2F%2FARAAGgwyMjA3NDg3MTQ4NjMiDN8a%2B5gze9YE%2Ft%2BMrSrXAzhAe4GZoAQW7VNYDegEgTrvoUpm%2F9LoX0GLQPp%2Bu3SsceQJhegdpJXvK4G5XV1wKBSYmdXT0lNA%2FJCszMuYezMtdeO5yoIQmzJZIO1%2FU7WGTtBcMujHuLkSMwOyafnCqGP11fWLwPtGx1X8nDSln8F3Q28DqmRxt0g%2FXMk35jTLSIAFBX7k%2FrNk2UhN0xx%2Fuy4DIG8cvr0dKUDE0pA7spuHkS8tnCgX4QU8byhYkGCXx1sOkNCjNh0nVJbv8WAJTVs4HPo%2FEcQkaeftYVgQ%2FjodjHFJDvkMFfbjHE8666IwFlt4Zqr5YJ%2FQ5z8DCl1UbV7sJ8vIvAcYdzUI46JYWvhmRFOo4M35JfYNJRrs4wizUb0krl7DA4WTRdJ0EZXQ1y8%2FRtrYTeysE22UUUw%2BK3wcrZULRXCmjKWHskE%2FVg6xEdTXufTRyAWwcvoGV13LfdNh6jzL5J6J1E5qkKiq%2BMjM1W9fZi%2FL9dv8xwcyw%2FH6TmmBeUyskD8BjrLUkRNDRY9%2B4LYWoNIZqSj69E39CIrfhT5%2B1Kq%2Betp3f5%2FG0NWXDgFx5YVzgkP8ng%2FCRefBL%2F9I%2BbYR5X0%2B9JljJ6L21401%2FELZzBqCOkcaHisszBnOb7tCSA96rjCxhPSQBjqlAQWRXFnzpgBYc1JKv0PQzR3gJqobM965KFlESm7EORYSDeYQAJJeYU3%2BMcaXBNcwuKjQS0PGBrXURXV8QXfCDp7YuEOis5LV6mQRv5YRASu6gbT1nQsgzV1diEkVh%2FXKG6QyzqWL3P1waeTrpQoO8lmfU%2F15ryvHmxGxR8AQmG0TW5up9QWFo0zSkJsGhxjDvclITIIFuatwc9ua9uCzJein%2FpJpXA%3D%3D&Expires=1646201235 [following]\n",
            "--2022-02-28 18:07:15--  https://encode-public.s3.amazonaws.com/2016/05/30/81e75b9d-f571-43ef-b644-2cf4dfb2ddbc/ENCFF726ZWD.fastq.gz?response-content-disposition=attachment%3B%20filename%3DENCFF726ZWD.fastq.gz&AWSAccessKeyId=ASIATGZNGCNXZ432VCC3&Signature=QR821mJ9xHPZvn8Uk%2Fo2cbRiVXk%3D&x-amz-security-token=IQoJb3JpZ2luX2VjEJn%2F%2F%2F%2F%2F%2F%2F%2F%2F%2FwEaCXVzLXdlc3QtMiJHMEUCIQDmnJF9590s83BJbzBdCqsc3RBzn2enFgxNa4CnPNEJ3QIgD15O8os5NzmgeTmlTFbBi4vbTeFEsjwgF9BkiIUgeR0qgwQI8v%2F%2F%2F%2F%2F%2F%2F%2F%2F%2FARAAGgwyMjA3NDg3MTQ4NjMiDN8a%2B5gze9YE%2Ft%2BMrSrXAzhAe4GZoAQW7VNYDegEgTrvoUpm%2F9LoX0GLQPp%2Bu3SsceQJhegdpJXvK4G5XV1wKBSYmdXT0lNA%2FJCszMuYezMtdeO5yoIQmzJZIO1%2FU7WGTtBcMujHuLkSMwOyafnCqGP11fWLwPtGx1X8nDSln8F3Q28DqmRxt0g%2FXMk35jTLSIAFBX7k%2FrNk2UhN0xx%2Fuy4DIG8cvr0dKUDE0pA7spuHkS8tnCgX4QU8byhYkGCXx1sOkNCjNh0nVJbv8WAJTVs4HPo%2FEcQkaeftYVgQ%2FjodjHFJDvkMFfbjHE8666IwFlt4Zqr5YJ%2FQ5z8DCl1UbV7sJ8vIvAcYdzUI46JYWvhmRFOo4M35JfYNJRrs4wizUb0krl7DA4WTRdJ0EZXQ1y8%2FRtrYTeysE22UUUw%2BK3wcrZULRXCmjKWHskE%2FVg6xEdTXufTRyAWwcvoGV13LfdNh6jzL5J6J1E5qkKiq%2BMjM1W9fZi%2FL9dv8xwcyw%2FH6TmmBeUyskD8BjrLUkRNDRY9%2B4LYWoNIZqSj69E39CIrfhT5%2B1Kq%2Betp3f5%2FG0NWXDgFx5YVzgkP8ng%2FCRefBL%2F9I%2BbYR5X0%2B9JljJ6L21401%2FELZzBqCOkcaHisszBnOb7tCSA96rjCxhPSQBjqlAQWRXFnzpgBYc1JKv0PQzR3gJqobM965KFlESm7EORYSDeYQAJJeYU3%2BMcaXBNcwuKjQS0PGBrXURXV8QXfCDp7YuEOis5LV6mQRv5YRASu6gbT1nQsgzV1diEkVh%2FXKG6QyzqWL3P1waeTrpQoO8lmfU%2F15ryvHmxGxR8AQmG0TW5up9QWFo0zSkJsGhxjDvclITIIFuatwc9ua9uCzJein%2FpJpXA%3D%3D&Expires=1646201235\n",
            "Resolving encode-public.s3.amazonaws.com (encode-public.s3.amazonaws.com)... 52.218.253.139\n",
            "Connecting to encode-public.s3.amazonaws.com (encode-public.s3.amazonaws.com)|52.218.253.139|:443... connected.\n",
            "HTTP request sent, awaiting response... 200 OK\n",
            "Length: 868043168 (828M) [binary/octet-stream]\n",
            "Saving to: ‘ENCFF726ZWD.fastq.gz.1’\n",
            "\n",
            "ENCFF726ZWD.fastq.g 100%[===================>] 827.83M  18.4MB/s    in 46s     \n",
            "\n",
            "2022-02-28 18:08:03 (17.8 MB/s) - ‘ENCFF726ZWD.fastq.gz.1’ saved [868043168/868043168]\n",
            "\n",
            "--2022-02-28 18:08:03--  https://www.encodeproject.org/files/ENCFF612KDN/@@download/ENCFF612KDN.fastq.gz\n",
            "Resolving www.encodeproject.org (www.encodeproject.org)... 34.211.244.144\n",
            "Connecting to www.encodeproject.org (www.encodeproject.org)|34.211.244.144|:443... connected.\n",
            "HTTP request sent, awaiting response... 307 Temporary Redirect\n",
            "Location: https://encode-public.s3.amazonaws.com/2016/05/30/34cb85e8-26cf-4d97-90a7-dd2e1352e544/ENCFF612KDN.fastq.gz?response-content-disposition=attachment%3B%20filename%3DENCFF612KDN.fastq.gz&AWSAccessKeyId=ASIATGZNGCNXZ432VCC3&Signature=PHyiIJeRrMw06XWSrEYhYKzs6mg%3D&x-amz-security-token=IQoJb3JpZ2luX2VjEJn%2F%2F%2F%2F%2F%2F%2F%2F%2F%2FwEaCXVzLXdlc3QtMiJHMEUCIQDmnJF9590s83BJbzBdCqsc3RBzn2enFgxNa4CnPNEJ3QIgD15O8os5NzmgeTmlTFbBi4vbTeFEsjwgF9BkiIUgeR0qgwQI8v%2F%2F%2F%2F%2F%2F%2F%2F%2F%2FARAAGgwyMjA3NDg3MTQ4NjMiDN8a%2B5gze9YE%2Ft%2BMrSrXAzhAe4GZoAQW7VNYDegEgTrvoUpm%2F9LoX0GLQPp%2Bu3SsceQJhegdpJXvK4G5XV1wKBSYmdXT0lNA%2FJCszMuYezMtdeO5yoIQmzJZIO1%2FU7WGTtBcMujHuLkSMwOyafnCqGP11fWLwPtGx1X8nDSln8F3Q28DqmRxt0g%2FXMk35jTLSIAFBX7k%2FrNk2UhN0xx%2Fuy4DIG8cvr0dKUDE0pA7spuHkS8tnCgX4QU8byhYkGCXx1sOkNCjNh0nVJbv8WAJTVs4HPo%2FEcQkaeftYVgQ%2FjodjHFJDvkMFfbjHE8666IwFlt4Zqr5YJ%2FQ5z8DCl1UbV7sJ8vIvAcYdzUI46JYWvhmRFOo4M35JfYNJRrs4wizUb0krl7DA4WTRdJ0EZXQ1y8%2FRtrYTeysE22UUUw%2BK3wcrZULRXCmjKWHskE%2FVg6xEdTXufTRyAWwcvoGV13LfdNh6jzL5J6J1E5qkKiq%2BMjM1W9fZi%2FL9dv8xwcyw%2FH6TmmBeUyskD8BjrLUkRNDRY9%2B4LYWoNIZqSj69E39CIrfhT5%2B1Kq%2Betp3f5%2FG0NWXDgFx5YVzgkP8ng%2FCRefBL%2F9I%2BbYR5X0%2B9JljJ6L21401%2FELZzBqCOkcaHisszBnOb7tCSA96rjCxhPSQBjqlAQWRXFnzpgBYc1JKv0PQzR3gJqobM965KFlESm7EORYSDeYQAJJeYU3%2BMcaXBNcwuKjQS0PGBrXURXV8QXfCDp7YuEOis5LV6mQRv5YRASu6gbT1nQsgzV1diEkVh%2FXKG6QyzqWL3P1waeTrpQoO8lmfU%2F15ryvHmxGxR8AQmG0TW5up9QWFo0zSkJsGhxjDvclITIIFuatwc9ua9uCzJein%2FpJpXA%3D%3D&Expires=1646201283 [following]\n",
            "--2022-02-28 18:08:04--  https://encode-public.s3.amazonaws.com/2016/05/30/34cb85e8-26cf-4d97-90a7-dd2e1352e544/ENCFF612KDN.fastq.gz?response-content-disposition=attachment%3B%20filename%3DENCFF612KDN.fastq.gz&AWSAccessKeyId=ASIATGZNGCNXZ432VCC3&Signature=PHyiIJeRrMw06XWSrEYhYKzs6mg%3D&x-amz-security-token=IQoJb3JpZ2luX2VjEJn%2F%2F%2F%2F%2F%2F%2F%2F%2F%2FwEaCXVzLXdlc3QtMiJHMEUCIQDmnJF9590s83BJbzBdCqsc3RBzn2enFgxNa4CnPNEJ3QIgD15O8os5NzmgeTmlTFbBi4vbTeFEsjwgF9BkiIUgeR0qgwQI8v%2F%2F%2F%2F%2F%2F%2F%2F%2F%2FARAAGgwyMjA3NDg3MTQ4NjMiDN8a%2B5gze9YE%2Ft%2BMrSrXAzhAe4GZoAQW7VNYDegEgTrvoUpm%2F9LoX0GLQPp%2Bu3SsceQJhegdpJXvK4G5XV1wKBSYmdXT0lNA%2FJCszMuYezMtdeO5yoIQmzJZIO1%2FU7WGTtBcMujHuLkSMwOyafnCqGP11fWLwPtGx1X8nDSln8F3Q28DqmRxt0g%2FXMk35jTLSIAFBX7k%2FrNk2UhN0xx%2Fuy4DIG8cvr0dKUDE0pA7spuHkS8tnCgX4QU8byhYkGCXx1sOkNCjNh0nVJbv8WAJTVs4HPo%2FEcQkaeftYVgQ%2FjodjHFJDvkMFfbjHE8666IwFlt4Zqr5YJ%2FQ5z8DCl1UbV7sJ8vIvAcYdzUI46JYWvhmRFOo4M35JfYNJRrs4wizUb0krl7DA4WTRdJ0EZXQ1y8%2FRtrYTeysE22UUUw%2BK3wcrZULRXCmjKWHskE%2FVg6xEdTXufTRyAWwcvoGV13LfdNh6jzL5J6J1E5qkKiq%2BMjM1W9fZi%2FL9dv8xwcyw%2FH6TmmBeUyskD8BjrLUkRNDRY9%2B4LYWoNIZqSj69E39CIrfhT5%2B1Kq%2Betp3f5%2FG0NWXDgFx5YVzgkP8ng%2FCRefBL%2F9I%2BbYR5X0%2B9JljJ6L21401%2FELZzBqCOkcaHisszBnOb7tCSA96rjCxhPSQBjqlAQWRXFnzpgBYc1JKv0PQzR3gJqobM965KFlESm7EORYSDeYQAJJeYU3%2BMcaXBNcwuKjQS0PGBrXURXV8QXfCDp7YuEOis5LV6mQRv5YRASu6gbT1nQsgzV1diEkVh%2FXKG6QyzqWL3P1waeTrpQoO8lmfU%2F15ryvHmxGxR8AQmG0TW5up9QWFo0zSkJsGhxjDvclITIIFuatwc9ua9uCzJein%2FpJpXA%3D%3D&Expires=1646201283\n",
            "Resolving encode-public.s3.amazonaws.com (encode-public.s3.amazonaws.com)... 52.218.168.83\n",
            "Connecting to encode-public.s3.amazonaws.com (encode-public.s3.amazonaws.com)|52.218.168.83|:443... connected.\n",
            "HTTP request sent, awaiting response... 200 OK\n",
            "Length: 610073369 (582M) [binary/octet-stream]\n",
            "Saving to: ‘ENCFF612KDN.fastq.gz’\n",
            "\n",
            "ENCFF612KDN.fastq.g 100%[===================>] 581.81M  18.8MB/s    in 32s     \n",
            "\n",
            "2022-02-28 18:08:37 (17.9 MB/s) - ‘ENCFF612KDN.fastq.gz’ saved [610073369/610073369]\n",
            "\n"
          ]
        }
      ],
      "source": [
        "# ChIP-seq на гистоновой метке\n",
        "!wget https://www.encodeproject.org/files/ENCFF052DPD/@@download/ENCFF052DPD.fastq.gz\n",
        "!wget https://www.encodeproject.org/files/ENCFF726ZWD/@@download/ENCFF726ZWD.fastq.gz\n",
        "# ChIP-seq контроль\n",
        "!wget https://www.encodeproject.org/files/ENCFF612KDN/@@download/ENCFF612KDN.fastq.gz"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "id": "v73fH7WiVDu9"
      },
      "outputs": [],
      "source": [
        "# распаковка ахрива\n",
        "!gzip -d /content/ENCFF052DPD.fastq.gz /content/ENCFF726ZWD.fastq.gz /content/ENCFF612KDN.fastq.gz"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "6sPcIefNHx-L"
      },
      "source": [
        "## FastQC"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "5XM6jBcuGjMV",
        "outputId": "b2293043-7c54-4603-f8d1-70c3406884aa"
      },
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Started analysis of ENCFF052DPD.fastq\n",
            "Approx 5% complete for ENCFF052DPD.fastq\n",
            "Approx 10% complete for ENCFF052DPD.fastq\n",
            "Approx 15% complete for ENCFF052DPD.fastq\n",
            "Approx 20% complete for ENCFF052DPD.fastq\n",
            "Approx 25% complete for ENCFF052DPD.fastq\n",
            "Approx 30% complete for ENCFF052DPD.fastq\n",
            "Approx 35% complete for ENCFF052DPD.fastq\n",
            "Approx 40% complete for ENCFF052DPD.fastq\n",
            "Approx 45% complete for ENCFF052DPD.fastq\n",
            "Approx 50% complete for ENCFF052DPD.fastq\n",
            "Approx 55% complete for ENCFF052DPD.fastq\n",
            "Approx 60% complete for ENCFF052DPD.fastq\n",
            "Approx 65% complete for ENCFF052DPD.fastq\n",
            "Approx 70% complete for ENCFF052DPD.fastq\n",
            "Approx 75% complete for ENCFF052DPD.fastq\n",
            "Approx 80% complete for ENCFF052DPD.fastq\n",
            "Approx 85% complete for ENCFF052DPD.fastq\n",
            "Approx 90% complete for ENCFF052DPD.fastq\n",
            "Approx 95% complete for ENCFF052DPD.fastq\n",
            "Analysis complete for ENCFF052DPD.fastq\n",
            "Started analysis of ENCFF726ZWD.fastq\n",
            "Approx 5% complete for ENCFF726ZWD.fastq\n",
            "Approx 10% complete for ENCFF726ZWD.fastq\n",
            "Approx 15% complete for ENCFF726ZWD.fastq\n",
            "Approx 20% complete for ENCFF726ZWD.fastq\n",
            "Approx 25% complete for ENCFF726ZWD.fastq\n",
            "Approx 30% complete for ENCFF726ZWD.fastq\n",
            "Approx 35% complete for ENCFF726ZWD.fastq\n",
            "Approx 40% complete for ENCFF726ZWD.fastq\n",
            "Approx 45% complete for ENCFF726ZWD.fastq\n",
            "Approx 50% complete for ENCFF726ZWD.fastq\n",
            "Approx 55% complete for ENCFF726ZWD.fastq\n",
            "Approx 60% complete for ENCFF726ZWD.fastq\n",
            "Approx 65% complete for ENCFF726ZWD.fastq\n",
            "Approx 70% complete for ENCFF726ZWD.fastq\n",
            "Approx 75% complete for ENCFF726ZWD.fastq\n",
            "Approx 80% complete for ENCFF726ZWD.fastq\n",
            "Approx 85% complete for ENCFF726ZWD.fastq\n",
            "Approx 90% complete for ENCFF726ZWD.fastq\n",
            "Approx 95% complete for ENCFF726ZWD.fastq\n",
            "Analysis complete for ENCFF726ZWD.fastq\n",
            "Started analysis of ENCFF612KDN.fastq\n",
            "Approx 5% complete for ENCFF612KDN.fastq\n",
            "Approx 10% complete for ENCFF612KDN.fastq\n",
            "Approx 15% complete for ENCFF612KDN.fastq\n",
            "Approx 20% complete for ENCFF612KDN.fastq\n",
            "Approx 25% complete for ENCFF612KDN.fastq\n",
            "Approx 30% complete for ENCFF612KDN.fastq\n",
            "Approx 35% complete for ENCFF612KDN.fastq\n",
            "Approx 40% complete for ENCFF612KDN.fastq\n",
            "Approx 45% complete for ENCFF612KDN.fastq\n",
            "Approx 50% complete for ENCFF612KDN.fastq\n",
            "Approx 55% complete for ENCFF612KDN.fastq\n",
            "Approx 60% complete for ENCFF612KDN.fastq\n",
            "Approx 65% complete for ENCFF612KDN.fastq\n",
            "Approx 70% complete for ENCFF612KDN.fastq\n",
            "Approx 75% complete for ENCFF612KDN.fastq\n",
            "Approx 80% complete for ENCFF612KDN.fastq\n",
            "Approx 85% complete for ENCFF612KDN.fastq\n",
            "Approx 90% complete for ENCFF612KDN.fastq\n",
            "Approx 95% complete for ENCFF612KDN.fastq\n",
            "Analysis complete for ENCFF612KDN.fastq\n"
          ]
        }
      ],
      "source": [
        "!./FastQC/fastqc /content/ENCFF052DPD.fastq\n",
        "!./FastQC/fastqc /content/ENCFF726ZWD.fastq\n",
        "!./FastQC/fastqc /content/ENCFF612KDN.fastq"
      ]
    },
    {
      "cell_type": "markdown",
      "source": [
        "***\n",
        "\n",
        "Анализ FastQC см. в README.\n",
        "\n",
        "Дальнейшее подрезание прочтений по результатам анализа не потребовалось.\n",
        "\n",
        "***"
      ],
      "metadata": {
        "id": "a06HC3mmNG2u"
      }
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "W9QHWBLlpMjQ"
      },
      "source": [
        "## Выравнивание на хромосому\n",
        "\n",
        "Следует выбрать одну хромосому, потому что ресурсы google colab ограничены. В случае проблем, таких как Segmentation fault, error 139, следует выбрать хромосому меньшего размера.\n",
        "\n",
        "Последовательность нуклеотидов хромосомы можно скачать по адресу https://hgdownload.soe.ucsc.edu/goldenPath/hg38/chromosomes/"
      ]
    },
    {
      "cell_type": "markdown",
      "source": [
        "Возьмем 17 хромосому для выравнивания"
      ],
      "metadata": {
        "id": "jF-k2PHnRQds"
      }
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "N4JIGZOsGiY2",
        "outputId": "a95fb7f0-6b44-40f3-d795-3b9b86738e04"
      },
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "--2022-02-28 18:28:18--  https://hgdownload.soe.ucsc.edu/goldenPath/hg38/chromosomes/chr17.fa.gz\n",
            "Resolving hgdownload.soe.ucsc.edu (hgdownload.soe.ucsc.edu)... 128.114.119.163\n",
            "Connecting to hgdownload.soe.ucsc.edu (hgdownload.soe.ucsc.edu)|128.114.119.163|:443... connected.\n",
            "HTTP request sent, awaiting response... 200 OK\n",
            "Length: 25930986 (25M) [application/x-gzip]\n",
            "Saving to: ‘chr17.fa.gz’\n",
            "\n",
            "chr17.fa.gz         100%[===================>]  24.73M  10.0MB/s    in 2.5s    \n",
            "\n",
            "2022-02-28 18:28:22 (10.0 MB/s) - ‘chr17.fa.gz’ saved [25930986/25930986]\n",
            "\n"
          ]
        }
      ],
      "source": [
        "!wget https://hgdownload.soe.ucsc.edu/goldenPath/hg38/chromosomes/chr17.fa.gz\n",
        "!gzip -d chr17.fa.gz"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "bhraQ6NdGiPN",
        "outputId": "ee2ce459-36d3-4ac3-d91c-2f973d60f897"
      },
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Settings:\n",
            "  Output files: \"chromosome_index.*.bt2\"\n",
            "  Line rate: 6 (line is 64 bytes)\n",
            "  Lines per side: 1 (side is 64 bytes)\n",
            "  Offset rate: 4 (one in 16)\n",
            "  FTable chars: 10\n",
            "  Strings: unpacked\n",
            "  Max bucket size: default\n",
            "  Max bucket size, sqrt multiplier: default\n",
            "  Max bucket size, len divisor: 4\n",
            "  Difference-cover sample period: 1024\n",
            "  Endianness: little\n",
            "  Actual local endianness: little\n",
            "  Sanity checking: disabled\n",
            "  Assertions: disabled\n",
            "  Random seed: 0\n",
            "  Sizeofs: void*:8, int:4, long:8, size_t:8\n",
            "Input files DNA, FASTA:\n",
            "  chr17.fa\n",
            "Building a SMALL index\n",
            "Reading reference sizes\n",
            "  Time reading reference sizes: 00:00:01\n",
            "Calculating joined length\n",
            "Writing header\n",
            "Reserving space for joined string\n",
            "Joining reference sequences\n",
            "  Time to join reference sequences: 00:00:00\n",
            "bmax according to bmaxDivN setting: 20730051\n",
            "Using parameters --bmax 15547539 --dcv 1024\n",
            "  Doing ahead-of-time memory usage test\n",
            "  Passed!  Constructing with these parameters: --bmax 15547539 --dcv 1024\n",
            "Constructing suffix-array element generator\n",
            "Building DifferenceCoverSample\n",
            "  Building sPrime\n",
            "  Building sPrimeOrder\n",
            "  V-Sorting samples\n",
            "  V-Sorting samples time: 00:00:02\n",
            "  Allocating rank array\n",
            "  Ranking v-sort output\n",
            "  Ranking v-sort output time: 00:00:00\n",
            "  Invoking Larsson-Sadakane on ranks\n",
            "  Invoking Larsson-Sadakane on ranks time: 00:00:01\n",
            "  Sanity-checking and returning\n",
            "Building samples\n",
            "Reserving space for 12 sample suffixes\n",
            "Generating random suffixes\n",
            "QSorting 12 sample offsets, eliminating duplicates\n",
            "QSorting sample offsets, eliminating duplicates time: 00:00:00\n",
            "Multikey QSorting 12 samples\n",
            "  (Using difference cover)\n",
            "  Multikey QSorting samples time: 00:00:00\n",
            "Calculating bucket sizes\n",
            "Splitting and merging\n",
            "  Splitting and merging time: 00:00:00\n",
            "Avg bucket size: 8.29202e+07 (target: 15547538)\n",
            "Converting suffix-array elements to index image\n",
            "Allocating ftab, absorbFtab\n",
            "Entering Ebwt loop\n",
            "Getting block 1 of 1\n",
            "  No samples; assembling all-inclusive block\n",
            "  Sorting block of length 82920204 for bucket 1\n",
            "  (Using difference cover)\n",
            "  Sorting block time: 00:00:29\n",
            "Returning block of 82920205 for bucket 1\n",
            "Exited Ebwt loop\n",
            "fchr[A]: 0\n",
            "fchr[C]: 22639499\n",
            "fchr[G]: 41363443\n",
            "fchr[T]: 60214943\n",
            "fchr[$]: 82920204\n",
            "Exiting Ebwt::buildToDisk()\n",
            "Returning from initFromVector\n",
            "Wrote 31835151 bytes to primary EBWT file: chromosome_index.1.bt2\n",
            "Wrote 20730056 bytes to secondary EBWT file: chromosome_index.2.bt2\n",
            "Re-opening _in1 and _in2 as input streams\n",
            "Returning from Ebwt constructor\n",
            "Headers:\n",
            "    len: 82920204\n",
            "    bwtLen: 82920205\n",
            "    sz: 20730051\n",
            "    bwtSz: 20730052\n",
            "    lineRate: 6\n",
            "    offRate: 4\n",
            "    offMask: 0xfffffff0\n",
            "    ftabChars: 10\n",
            "    eftabLen: 20\n",
            "    eftabSz: 80\n",
            "    ftabLen: 1048577\n",
            "    ftabSz: 4194308\n",
            "    offsLen: 5182513\n",
            "    offsSz: 20730052\n",
            "    lineSz: 64\n",
            "    sideSz: 64\n",
            "    sideBwtSz: 48\n",
            "    sideBwtLen: 192\n",
            "    numSides: 431877\n",
            "    numLines: 431877\n",
            "    ebwtTotLen: 27640128\n",
            "    ebwtTotSz: 27640128\n",
            "    color: 0\n",
            "    reverse: 0\n",
            "Total time for call to driver() for forward index: 00:00:47\n",
            "Reading reference sizes\n",
            "  Time reading reference sizes: 00:00:00\n",
            "Calculating joined length\n",
            "Writing header\n",
            "Reserving space for joined string\n",
            "Joining reference sequences\n",
            "  Time to join reference sequences: 00:00:01\n",
            "  Time to reverse reference sequence: 00:00:00\n",
            "bmax according to bmaxDivN setting: 20730051\n",
            "Using parameters --bmax 15547539 --dcv 1024\n",
            "  Doing ahead-of-time memory usage test\n",
            "  Passed!  Constructing with these parameters: --bmax 15547539 --dcv 1024\n",
            "Constructing suffix-array element generator\n",
            "Building DifferenceCoverSample\n",
            "  Building sPrime\n",
            "  Building sPrimeOrder\n",
            "  V-Sorting samples\n",
            "  V-Sorting samples time: 00:00:02\n",
            "  Allocating rank array\n",
            "  Ranking v-sort output\n",
            "  Ranking v-sort output time: 00:00:00\n",
            "  Invoking Larsson-Sadakane on ranks\n",
            "  Invoking Larsson-Sadakane on ranks time: 00:00:01\n",
            "  Sanity-checking and returning\n",
            "Building samples\n",
            "Reserving space for 12 sample suffixes\n",
            "Generating random suffixes\n",
            "QSorting 12 sample offsets, eliminating duplicates\n",
            "QSorting sample offsets, eliminating duplicates time: 00:00:00\n",
            "Multikey QSorting 12 samples\n",
            "  (Using difference cover)\n",
            "  Multikey QSorting samples time: 00:00:00\n",
            "Calculating bucket sizes\n",
            "Splitting and merging\n",
            "  Splitting and merging time: 00:00:00\n",
            "Avg bucket size: 8.29202e+07 (target: 15547538)\n",
            "Converting suffix-array elements to index image\n",
            "Allocating ftab, absorbFtab\n",
            "Entering Ebwt loop\n",
            "Getting block 1 of 1\n",
            "  No samples; assembling all-inclusive block\n",
            "  Sorting block of length 82920204 for bucket 1\n",
            "  (Using difference cover)\n",
            "  Sorting block time: 00:00:29\n",
            "Returning block of 82920205 for bucket 1\n",
            "Exited Ebwt loop\n",
            "fchr[A]: 0\n",
            "fchr[C]: 22639499\n",
            "fchr[G]: 41363443\n",
            "fchr[T]: 60214943\n",
            "fchr[$]: 82920204\n",
            "Exiting Ebwt::buildToDisk()\n",
            "Returning from initFromVector\n",
            "Wrote 31835151 bytes to primary EBWT file: chromosome_index.rev.1.bt2\n",
            "Wrote 20730056 bytes to secondary EBWT file: chromosome_index.rev.2.bt2\n",
            "Re-opening _in1 and _in2 as input streams\n",
            "Returning from Ebwt constructor\n",
            "Headers:\n",
            "    len: 82920204\n",
            "    bwtLen: 82920205\n",
            "    sz: 20730051\n",
            "    bwtSz: 20730052\n",
            "    lineRate: 6\n",
            "    offRate: 4\n",
            "    offMask: 0xfffffff0\n",
            "    ftabChars: 10\n",
            "    eftabLen: 20\n",
            "    eftabSz: 80\n",
            "    ftabLen: 1048577\n",
            "    ftabSz: 4194308\n",
            "    offsLen: 5182513\n",
            "    offsSz: 20730052\n",
            "    lineSz: 64\n",
            "    sideSz: 64\n",
            "    sideBwtSz: 48\n",
            "    sideBwtLen: 192\n",
            "    numSides: 431877\n",
            "    numLines: 431877\n",
            "    ebwtTotLen: 27640128\n",
            "    ebwtTotSz: 27640128\n",
            "    color: 0\n",
            "    reverse: 1\n",
            "Total time for backward call to driver() for mirror index: 00:00:47\n",
            "CPU times: user 710 ms, sys: 112 ms, total: 822 ms\n",
            "Wall time: 1min 33s\n"
          ]
        }
      ],
      "source": [
        "%%time\n",
        "!bowtie2-build chr17.fa chromosome_index"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "id": "E_usI8XUrQnR"
      },
      "outputs": [],
      "source": [
        "!mkdir bowtie2_res"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "id": "3t80UATHqFe7",
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "outputId": "7c86fd1c-a92c-4dc0-f69f-e39a98f95664"
      },
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "30927779 reads; of these:\n",
            "  30927779 (100.00%) were unpaired; of these:\n",
            "    25696380 (83.09%) aligned 0 times\n",
            "    1438430 (4.65%) aligned exactly 1 time\n",
            "    3792969 (12.26%) aligned >1 times\n",
            "16.91% overall alignment rate\n",
            "CPU times: user 8.39 s, sys: 1.16 s, total: 9.54 s\n",
            "Wall time: 20min 32s\n"
          ]
        }
      ],
      "source": [
        "# 1 реплика\n",
        "%%time\n",
        "!bowtie2  -p 2 \\\n",
        "          -x chromosome_index \\\n",
        "          -U ENCFF052DPD.fastq \\\n",
        "          -S bowtie2_res/res_DPD.sam"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "id": "so1jTdGHoEir",
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "outputId": "3e1f4970-f121-4bef-e743-9add06d1bbed"
      },
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "30504189 reads; of these:\n",
            "  30504189 (100.00%) were unpaired; of these:\n",
            "    25698139 (84.24%) aligned 0 times\n",
            "    1480427 (4.85%) aligned exactly 1 time\n",
            "    3325623 (10.90%) aligned >1 times\n",
            "15.76% overall alignment rate\n",
            "CPU times: user 7.96 s, sys: 1.09 s, total: 9.05 s\n",
            "Wall time: 19min 11s\n"
          ]
        }
      ],
      "source": [
        "# 2 реплика\n",
        "%%time\n",
        "!bowtie2  -p 2 \\\n",
        "          -x chromosome_index \\\n",
        "          -U ENCFF726ZWD.fastq \\\n",
        "          -S bowtie2_res/res_ZWD.sam"
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "# контроль\n",
        "%%time\n",
        "!bowtie2  -p 2 \\\n",
        "          -x chromosome_index \\\n",
        "          -U ENCFF612KDN.fastq \\\n",
        "          -S bowtie2_res/res_KDN.sam"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "l4PxypR6SjZP",
        "outputId": "ca79985c-9bfc-4060-e07b-5d8d6f8ab71e"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "20663767 reads; of these:\n",
            "  20663767 (100.00%) were unpaired; of these:\n",
            "    17206299 (83.27%) aligned 0 times\n",
            "    995771 (4.82%) aligned exactly 1 time\n",
            "    2461697 (11.91%) aligned >1 times\n",
            "16.73% overall alignment rate\n",
            "CPU times: user 5.29 s, sys: 632 ms, total: 5.92 s\n",
            "Wall time: 12min 43s\n"
          ]
        }
      ]
    },
    {
      "cell_type": "markdown",
      "source": [
        "### Таблица со статистикой по выравниванию"
      ],
      "metadata": {
        "id": "jZJiaaJBZBc7"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "import pandas as pd"
      ],
      "metadata": {
        "id": "RvQFuOhMXcw3"
      },
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "data = {'Общее число ридов':[30927779,\n",
        "                             30504189,\n",
        "                             20663767],\n",
        "        'Выровнившиеся уникально':['1438430',\n",
        "                                   '1480427',\n",
        "                                   '995771'],\n",
        "        'Выровнившиеся уникально (%)':['4.65%', \n",
        "                                       '4.85%',\n",
        "                                       '4.82%'],\n",
        "        'Выровнившиеся неуникально':['3792969',\n",
        "                                     '3325623',\n",
        "                                     '2461697'],\n",
        "        'Выровнившиеся неуникально (%)':['12.26%',\n",
        "                                         '10.90%',\n",
        "                                         '11.91%'],\n",
        "        'Не выровнившиеся':['25696380',\n",
        "                            '25698139',\n",
        "                            '17206299'],\n",
        "        'Не выровнившиеся (%)':['83.09%',\n",
        "                                '84.24%',\n",
        "                                '83.27%']}\n",
        "\n",
        "df = pd.DataFrame(data, index =['ENCFF052DPD',\n",
        "                                'ENCFF726ZWD',\n",
        "                                'ENCFF612KDN'])\n",
        "df"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 242
        },
        "id": "zp118ckhXffz",
        "outputId": "72e641b5-e637-4bbc-807d-d7eff8c3aa4d"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/html": [
              "\n",
              "  <div id=\"df-c288af3a-0bb9-4972-b87f-a563bb19f905\">\n",
              "    <div class=\"colab-df-container\">\n",
              "      <div>\n",
              "<style scoped>\n",
              "    .dataframe tbody tr th:only-of-type {\n",
              "        vertical-align: middle;\n",
              "    }\n",
              "\n",
              "    .dataframe tbody tr th {\n",
              "        vertical-align: top;\n",
              "    }\n",
              "\n",
              "    .dataframe thead th {\n",
              "        text-align: right;\n",
              "    }\n",
              "</style>\n",
              "<table border=\"1\" class=\"dataframe\">\n",
              "  <thead>\n",
              "    <tr style=\"text-align: right;\">\n",
              "      <th></th>\n",
              "      <th>Общее число ридов</th>\n",
              "      <th>Выровнившиеся уникально</th>\n",
              "      <th>Выровнившиеся уникально (%)</th>\n",
              "      <th>Выровнившиеся неуникально</th>\n",
              "      <th>Выровнившиеся неуникально (%)</th>\n",
              "      <th>Не выровнившиеся</th>\n",
              "      <th>Не выровнившиеся (%)</th>\n",
              "    </tr>\n",
              "  </thead>\n",
              "  <tbody>\n",
              "    <tr>\n",
              "      <th>ENCFF052DPD</th>\n",
              "      <td>30927779</td>\n",
              "      <td>1438430</td>\n",
              "      <td>4.65%</td>\n",
              "      <td>3792969</td>\n",
              "      <td>12.26%</td>\n",
              "      <td>25696380</td>\n",
              "      <td>83.09%</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>ENCFF726ZWD</th>\n",
              "      <td>30504189</td>\n",
              "      <td>1480427</td>\n",
              "      <td>4.85%</td>\n",
              "      <td>3325623</td>\n",
              "      <td>10.90%</td>\n",
              "      <td>25698139</td>\n",
              "      <td>84.24%</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>ENCFF612KDN</th>\n",
              "      <td>20663767</td>\n",
              "      <td>995771</td>\n",
              "      <td>4.82%</td>\n",
              "      <td>2461697</td>\n",
              "      <td>11.91%</td>\n",
              "      <td>17206299</td>\n",
              "      <td>83.27%</td>\n",
              "    </tr>\n",
              "  </tbody>\n",
              "</table>\n",
              "</div>\n",
              "      <button class=\"colab-df-convert\" onclick=\"convertToInteractive('df-c288af3a-0bb9-4972-b87f-a563bb19f905')\"\n",
              "              title=\"Convert this dataframe to an interactive table.\"\n",
              "              style=\"display:none;\">\n",
              "        \n",
              "  <svg xmlns=\"http://www.w3.org/2000/svg\" height=\"24px\"viewBox=\"0 0 24 24\"\n",
              "       width=\"24px\">\n",
              "    <path d=\"M0 0h24v24H0V0z\" fill=\"none\"/>\n",
              "    <path d=\"M18.56 5.44l.94 2.06.94-2.06 2.06-.94-2.06-.94-.94-2.06-.94 2.06-2.06.94zm-11 1L8.5 8.5l.94-2.06 2.06-.94-2.06-.94L8.5 2.5l-.94 2.06-2.06.94zm10 10l.94 2.06.94-2.06 2.06-.94-2.06-.94-.94-2.06-.94 2.06-2.06.94z\"/><path d=\"M17.41 7.96l-1.37-1.37c-.4-.4-.92-.59-1.43-.59-.52 0-1.04.2-1.43.59L10.3 9.45l-7.72 7.72c-.78.78-.78 2.05 0 2.83L4 21.41c.39.39.9.59 1.41.59.51 0 1.02-.2 1.41-.59l7.78-7.78 2.81-2.81c.8-.78.8-2.07 0-2.86zM5.41 20L4 18.59l7.72-7.72 1.47 1.35L5.41 20z\"/>\n",
              "  </svg>\n",
              "      </button>\n",
              "      \n",
              "  <style>\n",
              "    .colab-df-container {\n",
              "      display:flex;\n",
              "      flex-wrap:wrap;\n",
              "      gap: 12px;\n",
              "    }\n",
              "\n",
              "    .colab-df-convert {\n",
              "      background-color: #E8F0FE;\n",
              "      border: none;\n",
              "      border-radius: 50%;\n",
              "      cursor: pointer;\n",
              "      display: none;\n",
              "      fill: #1967D2;\n",
              "      height: 32px;\n",
              "      padding: 0 0 0 0;\n",
              "      width: 32px;\n",
              "    }\n",
              "\n",
              "    .colab-df-convert:hover {\n",
              "      background-color: #E2EBFA;\n",
              "      box-shadow: 0px 1px 2px rgba(60, 64, 67, 0.3), 0px 1px 3px 1px rgba(60, 64, 67, 0.15);\n",
              "      fill: #174EA6;\n",
              "    }\n",
              "\n",
              "    [theme=dark] .colab-df-convert {\n",
              "      background-color: #3B4455;\n",
              "      fill: #D2E3FC;\n",
              "    }\n",
              "\n",
              "    [theme=dark] .colab-df-convert:hover {\n",
              "      background-color: #434B5C;\n",
              "      box-shadow: 0px 1px 3px 1px rgba(0, 0, 0, 0.15);\n",
              "      filter: drop-shadow(0px 1px 2px rgba(0, 0, 0, 0.3));\n",
              "      fill: #FFFFFF;\n",
              "    }\n",
              "  </style>\n",
              "\n",
              "      <script>\n",
              "        const buttonEl =\n",
              "          document.querySelector('#df-c288af3a-0bb9-4972-b87f-a563bb19f905 button.colab-df-convert');\n",
              "        buttonEl.style.display =\n",
              "          google.colab.kernel.accessAllowed ? 'block' : 'none';\n",
              "\n",
              "        async function convertToInteractive(key) {\n",
              "          const element = document.querySelector('#df-c288af3a-0bb9-4972-b87f-a563bb19f905');\n",
              "          const dataTable =\n",
              "            await google.colab.kernel.invokeFunction('convertToInteractive',\n",
              "                                                     [key], {});\n",
              "          if (!dataTable) return;\n",
              "\n",
              "          const docLinkHtml = 'Like what you see? Visit the ' +\n",
              "            '<a target=\"_blank\" href=https://colab.research.google.com/notebooks/data_table.ipynb>data table notebook</a>'\n",
              "            + ' to learn more about interactive tables.';\n",
              "          element.innerHTML = '';\n",
              "          dataTable['output_type'] = 'display_data';\n",
              "          await google.colab.output.renderOutput(dataTable, element);\n",
              "          const docLink = document.createElement('div');\n",
              "          docLink.innerHTML = docLinkHtml;\n",
              "          element.appendChild(docLink);\n",
              "        }\n",
              "      </script>\n",
              "    </div>\n",
              "  </div>\n",
              "  "
            ],
            "text/plain": [
              "             Общее число ридов  ... Не выровнившиеся (%)\n",
              "ENCFF052DPD           30927779  ...               83.09%\n",
              "ENCFF726ZWD           30504189  ...               84.24%\n",
              "ENCFF612KDN           20663767  ...               83.27%\n",
              "\n",
              "[3 rows x 7 columns]"
            ]
          },
          "metadata": {},
          "execution_count": 31
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "df.to_csv('stats.csv')"
      ],
      "metadata": {
        "id": "huJ6P_NmZFtO"
      },
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "ysoDrnSCrei0"
      },
      "source": [
        "Проанализируйте выдачу bowtie. Почему процент выравниваний получился именно таким?\n",
        "\n",
        "***\n",
        "\n",
        "**Ответ:** потому что мы выравнивали риды на одну хромосому, которая составляет мАлую часть генома человека, ведь там всего 23 пары хромосом. Поэтому процент выравниваний и получился таким низким.\n",
        "\n",
        "***\n",
        "\n",
        "Имеет смысл для дальнейшего анализа отобрать уникально картированные риды."
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "TmfJKT0rmLyp"
      },
      "source": [
        "Преобразование .sam в .bam (не требуется если запускать macs2 на .sam файлах)"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "id": "2PteJ-KVqFhs"
      },
      "outputs": [],
      "source": [
        "!samtools view -h -S -b -o bowtie2_res/res_DPD.bam bowtie2_res/res_DPD.sam\n",
        "!samtools view -h -S -b -o bowtie2_res/res_ZWD.bam bowtie2_res/res_ZWD.sam\n",
        "!samtools view -h -S -b -o bowtie2_res/res_KDN.bam bowtie2_res/res_KDN.sam"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "eLtG7xtRup19"
      },
      "source": [
        "На этом моменте можно скачать себе на компьютер все полученные .bam или .sam файлы на случай сбоя."
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "XZtQrPZMmgnC"
      },
      "source": [
        "## Peak calling\n",
        "\n",
        "Помимо macs2 может быть использован пакет Homer для поиска пиков, а также macs3."
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "id": "uX1ps4VDp_Ae"
      },
      "outputs": [],
      "source": [
        "!mkdir macs2"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "id": "EJQOCcv2nJW_",
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "outputId": "01dca69d-4638-48ee-f5f2-34674db0d2bb"
      },
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "INFO  @ Mon, 28 Feb 2022 19:45:35: \n",
            "# Command line: callpeak --broad -t bowtie2_res/res_DPD.sam -c bowtie2_res/res_KDN.sam -f SAM --outdir macs2 -n ENCFF052DPD\n",
            "# ARGUMENTS LIST:\n",
            "# name = ENCFF052DPD\n",
            "# format = SAM\n",
            "# ChIP-seq file = ['bowtie2_res/res_DPD.sam']\n",
            "# control file = ['bowtie2_res/res_KDN.sam']\n",
            "# effective genome size = 2.70e+09\n",
            "# band width = 300\n",
            "# model fold = [5, 50]\n",
            "# qvalue cutoff for narrow/strong regions = 5.00e-02\n",
            "# qvalue cutoff for broad/weak regions = 1.00e-01\n",
            "# The maximum gap between significant sites is assigned as the read length/tag size.\n",
            "# The minimum length of peaks is assigned as the predicted fragment length \"d\".\n",
            "# Larger dataset will be scaled towards smaller dataset.\n",
            "# Range for calculating regional lambda is: 1000 bps and 10000 bps\n",
            "# Broad region calling is on\n",
            "# Paired-End mode is off\n",
            " \n",
            "INFO  @ Mon, 28 Feb 2022 19:45:35: #1 read tag files... \n",
            "INFO  @ Mon, 28 Feb 2022 19:45:35: #1 read treatment tags... \n",
            "INFO  @ Mon, 28 Feb 2022 19:45:40:  5000000 \n",
            "INFO  @ Mon, 28 Feb 2022 19:45:59:  23000000 \n",
            "INFO  @ Mon, 28 Feb 2022 19:46:01:  25000000 \n",
            "INFO  @ Mon, 28 Feb 2022 19:46:07: #1.2 read input tags... \n",
            "INFO  @ Mon, 28 Feb 2022 19:46:08:  1000000 \n",
            "INFO  @ Mon, 28 Feb 2022 19:46:13:  6000000 \n",
            "INFO  @ Mon, 28 Feb 2022 19:46:15:  8000000 \n",
            "INFO  @ Mon, 28 Feb 2022 19:46:17:  10000000 \n",
            "INFO  @ Mon, 28 Feb 2022 19:46:21:  14000000 \n",
            "INFO  @ Mon, 28 Feb 2022 19:46:22:  15000000 \n",
            "INFO  @ Mon, 28 Feb 2022 19:46:24:  17000000 \n",
            "INFO  @ Mon, 28 Feb 2022 19:46:26:  19000000 \n",
            "INFO  @ Mon, 28 Feb 2022 19:46:28: #1 tag size is determined as 51 bps \n",
            "INFO  @ Mon, 28 Feb 2022 19:46:28: #1 tag size = 51.0 \n",
            "INFO  @ Mon, 28 Feb 2022 19:46:28: #1  total tags in treatment: 5231399 \n",
            "INFO  @ Mon, 28 Feb 2022 19:46:28: #1 user defined the maximum tags... \n",
            "INFO  @ Mon, 28 Feb 2022 19:46:28: #1 filter out redundant tags at the same location and the same strand by allowing at most 1 tag(s) \n",
            "INFO  @ Mon, 28 Feb 2022 19:46:28: #1  tags after filtering in treatment: 4377343 \n",
            "INFO  @ Mon, 28 Feb 2022 19:46:28: #1  Redundant rate of treatment: 0.16 \n",
            "INFO  @ Mon, 28 Feb 2022 19:46:28: #1  total tags in control: 3457468 \n",
            "INFO  @ Mon, 28 Feb 2022 19:46:28: #1 user defined the maximum tags... \n",
            "INFO  @ Mon, 28 Feb 2022 19:46:28: #1 filter out redundant tags at the same location and the same strand by allowing at most 1 tag(s) \n",
            "INFO  @ Mon, 28 Feb 2022 19:46:28: #1  tags after filtering in control: 2971426 \n",
            "INFO  @ Mon, 28 Feb 2022 19:46:28: #1  Redundant rate of control: 0.14 \n",
            "INFO  @ Mon, 28 Feb 2022 19:46:28: #1 finished! \n",
            "INFO  @ Mon, 28 Feb 2022 19:46:28: #2 Build Peak Model... \n",
            "INFO  @ Mon, 28 Feb 2022 19:46:28: #2 looking for paired plus/minus strand peaks... \n",
            "INFO  @ Mon, 28 Feb 2022 19:46:30: #2 number of paired peaks: 44133 \n",
            "INFO  @ Mon, 28 Feb 2022 19:46:30: start model_add_line... \n",
            "INFO  @ Mon, 28 Feb 2022 19:46:30: start X-correlation... \n",
            "INFO  @ Mon, 28 Feb 2022 19:46:30: end of X-cor \n",
            "INFO  @ Mon, 28 Feb 2022 19:46:30: #2 finished! \n",
            "INFO  @ Mon, 28 Feb 2022 19:46:30: #2 predicted fragment length is 51 bps \n",
            "INFO  @ Mon, 28 Feb 2022 19:46:30: #2 alternative fragment length(s) may be 51 bps \n",
            "INFO  @ Mon, 28 Feb 2022 19:46:30: #2.2 Generate R script for model : macs2/ENCFF052DPD_model.r \n",
            "WARNING @ Mon, 28 Feb 2022 19:46:30: #2 Since the d (51) calculated from paired-peaks are smaller than 2*tag length, it may be influenced by unknown sequencing problem! \n",
            "WARNING @ Mon, 28 Feb 2022 19:46:30: #2 You may need to consider one of the other alternative d(s): 51 \n",
            "WARNING @ Mon, 28 Feb 2022 19:46:30: #2 You can restart the process with --nomodel --extsize XXX with your choice or an arbitrary number. Nontheless, MACS will continute computing. \n",
            "INFO  @ Mon, 28 Feb 2022 19:46:30: #3 Call peaks... \n",
            "INFO  @ Mon, 28 Feb 2022 19:46:30: #3 Call broad peaks with given level1 -log10qvalue cutoff and level2: 1.301030, 1.000000... \n",
            "INFO  @ Mon, 28 Feb 2022 19:46:30: #3 Pre-compute pvalue-qvalue table... \n",
            "INFO  @ Mon, 28 Feb 2022 19:46:43: #3 Call peaks for each chromosome... \n",
            "INFO  @ Mon, 28 Feb 2022 19:46:50: #4 Write output xls file... macs2/ENCFF052DPD_peaks.xls \n",
            "INFO  @ Mon, 28 Feb 2022 19:46:50: #4 Write broad peak in broadPeak format file... macs2/ENCFF052DPD_peaks.broadPeak \n",
            "INFO  @ Mon, 28 Feb 2022 19:46:50: #4 Write broad peak in bed12/gappedPeak format file... macs2/ENCFF052DPD_peaks.gappedPeak \n",
            "INFO  @ Mon, 28 Feb 2022 19:46:50: Done! \n"
          ]
        }
      ],
      "source": [
        "!macs2 callpeak --broad -t bowtie2_res/res_DPD.sam \\\n",
        "    -c bowtie2_res/res_KDN.sam \\\n",
        " \t  -f SAM \\\n",
        "\t  --outdir macs2 \\\n",
        "\t  -n ENCFF052DPD"
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "!macs2 callpeak --broad -t bowtie2_res/res_ZWD.sam \\\n",
        "    -c bowtie2_res/res_KDN.sam \\\n",
        " \t  -f SAM \\\n",
        "\t  --outdir macs2 \\\n",
        "      -n ENCFF726ZWD"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "UiL_pwRpfx7R",
        "outputId": "1ce72d56-d056-49e2-d345-0cafa1d80b94"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "INFO  @ Mon, 28 Feb 2022 19:46:51: \n",
            "# Command line: callpeak --broad -t bowtie2_res/res_ZWD.sam -c bowtie2_res/res_KDN.sam -f SAM --outdir macs2 -n ENCFF726ZWD\n",
            "# ARGUMENTS LIST:\n",
            "# name = ENCFF726ZWD\n",
            "# format = SAM\n",
            "# ChIP-seq file = ['bowtie2_res/res_ZWD.sam']\n",
            "# control file = ['bowtie2_res/res_KDN.sam']\n",
            "# effective genome size = 2.70e+09\n",
            "# band width = 300\n",
            "# model fold = [5, 50]\n",
            "# qvalue cutoff for narrow/strong regions = 5.00e-02\n",
            "# qvalue cutoff for broad/weak regions = 1.00e-01\n",
            "# The maximum gap between significant sites is assigned as the read length/tag size.\n",
            "# The minimum length of peaks is assigned as the predicted fragment length \"d\".\n",
            "# Larger dataset will be scaled towards smaller dataset.\n",
            "# Range for calculating regional lambda is: 1000 bps and 10000 bps\n",
            "# Broad region calling is on\n",
            "# Paired-End mode is off\n",
            " \n",
            "INFO  @ Mon, 28 Feb 2022 19:46:51: #1 read tag files... \n",
            "INFO  @ Mon, 28 Feb 2022 19:46:51: #1 read treatment tags... \n",
            "INFO  @ Mon, 28 Feb 2022 19:46:52:  1000000 \n",
            "INFO  @ Mon, 28 Feb 2022 19:46:53:  2000000 \n",
            "INFO  @ Mon, 28 Feb 2022 19:46:54:  3000000 \n",
            "INFO  @ Mon, 28 Feb 2022 19:46:59:  8000000 \n",
            "INFO  @ Mon, 28 Feb 2022 19:47:04:  13000000 \n",
            "INFO  @ Mon, 28 Feb 2022 19:47:11:  20000000 \n",
            "INFO  @ Mon, 28 Feb 2022 19:47:19:  28000000 \n",
            "INFO  @ Mon, 28 Feb 2022 19:47:22: #1.2 read input tags... \n",
            "INFO  @ Mon, 28 Feb 2022 19:47:23:  1000000 \n",
            "INFO  @ Mon, 28 Feb 2022 19:47:28:  6000000 \n",
            "INFO  @ Mon, 28 Feb 2022 19:47:30:  8000000 \n",
            "INFO  @ Mon, 28 Feb 2022 19:47:32:  10000000 \n",
            "INFO  @ Mon, 28 Feb 2022 19:47:36:  14000000 \n",
            "INFO  @ Mon, 28 Feb 2022 19:47:37:  15000000 \n",
            "INFO  @ Mon, 28 Feb 2022 19:47:39:  17000000 \n",
            "INFO  @ Mon, 28 Feb 2022 19:47:41:  19000000 \n",
            "INFO  @ Mon, 28 Feb 2022 19:47:43: #1 tag size is determined as 51 bps \n",
            "INFO  @ Mon, 28 Feb 2022 19:47:43: #1 tag size = 51.0 \n",
            "INFO  @ Mon, 28 Feb 2022 19:47:43: #1  total tags in treatment: 4806050 \n",
            "INFO  @ Mon, 28 Feb 2022 19:47:43: #1 user defined the maximum tags... \n",
            "INFO  @ Mon, 28 Feb 2022 19:47:43: #1 filter out redundant tags at the same location and the same strand by allowing at most 1 tag(s) \n",
            "INFO  @ Mon, 28 Feb 2022 19:47:43: #1  tags after filtering in treatment: 4144864 \n",
            "INFO  @ Mon, 28 Feb 2022 19:47:43: #1  Redundant rate of treatment: 0.14 \n",
            "INFO  @ Mon, 28 Feb 2022 19:47:43: #1  total tags in control: 3457468 \n",
            "INFO  @ Mon, 28 Feb 2022 19:47:43: #1 user defined the maximum tags... \n",
            "INFO  @ Mon, 28 Feb 2022 19:47:43: #1 filter out redundant tags at the same location and the same strand by allowing at most 1 tag(s) \n",
            "INFO  @ Mon, 28 Feb 2022 19:47:43: #1  tags after filtering in control: 2971426 \n",
            "INFO  @ Mon, 28 Feb 2022 19:47:43: #1  Redundant rate of control: 0.14 \n",
            "INFO  @ Mon, 28 Feb 2022 19:47:43: #1 finished! \n",
            "INFO  @ Mon, 28 Feb 2022 19:47:43: #2 Build Peak Model... \n",
            "INFO  @ Mon, 28 Feb 2022 19:47:43: #2 looking for paired plus/minus strand peaks... \n",
            "INFO  @ Mon, 28 Feb 2022 19:47:44: #2 number of paired peaks: 38001 \n",
            "INFO  @ Mon, 28 Feb 2022 19:47:44: start model_add_line... \n",
            "INFO  @ Mon, 28 Feb 2022 19:47:44: start X-correlation... \n",
            "INFO  @ Mon, 28 Feb 2022 19:47:44: end of X-cor \n",
            "INFO  @ Mon, 28 Feb 2022 19:47:44: #2 finished! \n",
            "INFO  @ Mon, 28 Feb 2022 19:47:44: #2 predicted fragment length is 51 bps \n",
            "INFO  @ Mon, 28 Feb 2022 19:47:44: #2 alternative fragment length(s) may be 51 bps \n",
            "INFO  @ Mon, 28 Feb 2022 19:47:44: #2.2 Generate R script for model : macs2/ENCFF726ZWD_model.r \n",
            "WARNING @ Mon, 28 Feb 2022 19:47:44: #2 Since the d (51) calculated from paired-peaks are smaller than 2*tag length, it may be influenced by unknown sequencing problem! \n",
            "WARNING @ Mon, 28 Feb 2022 19:47:44: #2 You may need to consider one of the other alternative d(s): 51 \n",
            "WARNING @ Mon, 28 Feb 2022 19:47:44: #2 You can restart the process with --nomodel --extsize XXX with your choice or an arbitrary number. Nontheless, MACS will continute computing. \n",
            "INFO  @ Mon, 28 Feb 2022 19:47:44: #3 Call peaks... \n",
            "INFO  @ Mon, 28 Feb 2022 19:47:44: #3 Call broad peaks with given level1 -log10qvalue cutoff and level2: 1.301030, 1.000000... \n",
            "INFO  @ Mon, 28 Feb 2022 19:47:44: #3 Pre-compute pvalue-qvalue table... \n",
            "INFO  @ Mon, 28 Feb 2022 19:47:57: #3 Call peaks for each chromosome... \n",
            "INFO  @ Mon, 28 Feb 2022 19:48:04: #4 Write output xls file... macs2/ENCFF726ZWD_peaks.xls \n",
            "INFO  @ Mon, 28 Feb 2022 19:48:04: #4 Write broad peak in broadPeak format file... macs2/ENCFF726ZWD_peaks.broadPeak \n",
            "INFO  @ Mon, 28 Feb 2022 19:48:04: #4 Write broad peak in bed12/gappedPeak format file... macs2/ENCFF726ZWD_peaks.gappedPeak \n",
            "INFO  @ Mon, 28 Feb 2022 19:48:04: Done! \n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "id": "HrNwxs7ArPz7",
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "outputId": "b35c7ab6-51d6-4fda-c876-c6f503ccd495"
      },
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "--2022-02-28 19:54:13--  https://www.encodeproject.org/files/ENCFF379KHF/@@download/ENCFF379KHF.bed.gz\n",
            "Resolving www.encodeproject.org (www.encodeproject.org)... 34.211.244.144\n",
            "Connecting to www.encodeproject.org (www.encodeproject.org)|34.211.244.144|:443... connected.\n",
            "HTTP request sent, awaiting response... 307 Temporary Redirect\n",
            "Location: https://encode-public.s3.amazonaws.com/2020/11/13/719403cd-c289-4dd1-b880-5a43b730e354/ENCFF379KHF.bed.gz?response-content-disposition=attachment%3B%20filename%3DENCFF379KHF.bed.gz&AWSAccessKeyId=ASIATGZNGCNXXZFOCYGP&Signature=qgop7DU9km0g8s4aJyaDIBwm3l0%3D&x-amz-security-token=IQoJb3JpZ2luX2VjEJv%2F%2F%2F%2F%2F%2F%2F%2F%2F%2FwEaCXVzLXdlc3QtMiJGMEQCIEbNEa13ARcL2Sn8Vf4cCHOoLLrrKemG0G96RgD6nGbkAiALKj9Ki44orcHZcsmPaLVCEJxH7DNnQ4Zk2AWs%2B4DveCqDBAj0%2F%2F%2F%2F%2F%2F%2F%2F%2F%2F8BEAAaDDIyMDc0ODcxNDg2MyIM4vsIXtC4l7JKnGytKtcD22hQXc%2FVFksjYvnHBPSLDE7ffhMC04a4051J4LKsi7xUvpkv3%2B74HVAO1flTLl2IDUbMAjAWyDA4ST2TH2UBvQELyIIduwuHpuBHqPNFe%2Fsu6EFddtxyk5%2FVCnICepDxPJbUXszkiYKK%2BscDxLaWhTUrwOg%2FHvMreQe%2FJpXpsGMAVr%2F3qRH5VP7ERHFI5w1JzpJRzV2tfaM%2FUsAyJ55GxnAWWFWsekqfNwouzSAbqJYZ8o9mUuVaLuFINHnZT27TVVdXUCVAxcMAk%2B980MnwMaVjJKZTtZx%2Bm1rX5KJVYdLBDnX9phRA7MN7py8FARy%2FHVIG0Rp79Iwl%2FLBVaKxxRVXBzrpwMRbTJXQkEK3FsyAAhylXfk%2FMMAix05HKKOJlE%2FlMy7HqRppYBYouztoSkFdG3Zi5km%2Bn6hXR%2FE%2FrCYFz%2FxPL%2B%2B1edh3mSqEt2ystuF27ES9Xf10AksH9X4WAjGKsBkzk%2BrdpQnC%2B0yF9JLn3DtExF4s7UDmM4zefr%2Fvogroh2JdUkkR1ipD%2FoD5MWbbZN2%2BACU2EhEU7GBbYyIhg6dWWunl2fCf7yGWulRw0NCaRxGEBohF%2F8TZV49juPcS0wxo6GZ8pSElALxeq8%2B7l4%2FG4wWOdMNG%2B9JAGOqYBTgeM5iZvQzCs6mRBeMsem0GZMPeiTWI1gNWdQSP%2FZFiVxGWzHK5EfIxbU0AkKtNtD%2F4qozMvQRny91aQaiLyIA0%2F0wFd4Ptbe%2B0kHFSIOYTDW%2BEJLAXtnJf6%2Fz%2BBKanKIHJTMFvXquaKO8zatLDGRYjMn5zTtQyYt2J7%2FnbQAp4H5kD4KNAn38BsNxjWrtu3qo5sUdBBaT2fD8rrhkRTyd5o%2BHRS3g%3D%3D&Expires=1646207654 [following]\n",
            "--2022-02-28 19:54:14--  https://encode-public.s3.amazonaws.com/2020/11/13/719403cd-c289-4dd1-b880-5a43b730e354/ENCFF379KHF.bed.gz?response-content-disposition=attachment%3B%20filename%3DENCFF379KHF.bed.gz&AWSAccessKeyId=ASIATGZNGCNXXZFOCYGP&Signature=qgop7DU9km0g8s4aJyaDIBwm3l0%3D&x-amz-security-token=IQoJb3JpZ2luX2VjEJv%2F%2F%2F%2F%2F%2F%2F%2F%2F%2FwEaCXVzLXdlc3QtMiJGMEQCIEbNEa13ARcL2Sn8Vf4cCHOoLLrrKemG0G96RgD6nGbkAiALKj9Ki44orcHZcsmPaLVCEJxH7DNnQ4Zk2AWs%2B4DveCqDBAj0%2F%2F%2F%2F%2F%2F%2F%2F%2F%2F8BEAAaDDIyMDc0ODcxNDg2MyIM4vsIXtC4l7JKnGytKtcD22hQXc%2FVFksjYvnHBPSLDE7ffhMC04a4051J4LKsi7xUvpkv3%2B74HVAO1flTLl2IDUbMAjAWyDA4ST2TH2UBvQELyIIduwuHpuBHqPNFe%2Fsu6EFddtxyk5%2FVCnICepDxPJbUXszkiYKK%2BscDxLaWhTUrwOg%2FHvMreQe%2FJpXpsGMAVr%2F3qRH5VP7ERHFI5w1JzpJRzV2tfaM%2FUsAyJ55GxnAWWFWsekqfNwouzSAbqJYZ8o9mUuVaLuFINHnZT27TVVdXUCVAxcMAk%2B980MnwMaVjJKZTtZx%2Bm1rX5KJVYdLBDnX9phRA7MN7py8FARy%2FHVIG0Rp79Iwl%2FLBVaKxxRVXBzrpwMRbTJXQkEK3FsyAAhylXfk%2FMMAix05HKKOJlE%2FlMy7HqRppYBYouztoSkFdG3Zi5km%2Bn6hXR%2FE%2FrCYFz%2FxPL%2B%2B1edh3mSqEt2ystuF27ES9Xf10AksH9X4WAjGKsBkzk%2BrdpQnC%2B0yF9JLn3DtExF4s7UDmM4zefr%2Fvogroh2JdUkkR1ipD%2FoD5MWbbZN2%2BACU2EhEU7GBbYyIhg6dWWunl2fCf7yGWulRw0NCaRxGEBohF%2F8TZV49juPcS0wxo6GZ8pSElALxeq8%2B7l4%2FG4wWOdMNG%2B9JAGOqYBTgeM5iZvQzCs6mRBeMsem0GZMPeiTWI1gNWdQSP%2FZFiVxGWzHK5EfIxbU0AkKtNtD%2F4qozMvQRny91aQaiLyIA0%2F0wFd4Ptbe%2B0kHFSIOYTDW%2BEJLAXtnJf6%2Fz%2BBKanKIHJTMFvXquaKO8zatLDGRYjMn5zTtQyYt2J7%2FnbQAp4H5kD4KNAn38BsNxjWrtu3qo5sUdBBaT2fD8rrhkRTyd5o%2BHRS3g%3D%3D&Expires=1646207654\n",
            "Resolving encode-public.s3.amazonaws.com (encode-public.s3.amazonaws.com)... 52.92.196.105\n",
            "Connecting to encode-public.s3.amazonaws.com (encode-public.s3.amazonaws.com)|52.92.196.105|:443... connected.\n",
            "HTTP request sent, awaiting response... 200 OK\n",
            "Length: 2442845 (2.3M) [binary/octet-stream]\n",
            "Saving to: ‘ENCFF379KHF.bed.gz’\n",
            "\n",
            "ENCFF379KHF.bed.gz  100%[===================>]   2.33M  2.08MB/s    in 1.1s    \n",
            "\n",
            "2022-02-28 19:54:16 (2.08 MB/s) - ‘ENCFF379KHF.bed.gz’ saved [2442845/2442845]\n",
            "\n"
          ]
        }
      ],
      "source": [
        "!wget https://www.encodeproject.org/files/ENCFF379KHF/@@download/ENCFF379KHF.bed.gz\n",
        "!gzip -d ENCFF379KHF.bed.gz"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "MYEGEbQjvEwY"
      },
      "source": [
        "## Сравнение результатов\n",
        "\n",
        "Сравниваем те пики, которые мы получили, с пиками, которые приведены в ENCODE (важно, чтобы версии генома hg38 или hg19) для .bed файла из ENCODE и той хромосомы, которую скачивали выше, сопадали.\n",
        "\n",
        "Проанализируйте полученные результаты и приведите свои рассуждения в README.md. Как можно объяснить различия в количестве пересечений?"
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "!mkdir venn1\n",
        "!mkdir venn2\n",
        "!mkdir venn3\n",
        "!mkdir venn4"
      ],
      "metadata": {
        "id": "zQbSWH-tMtZz"
      },
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "id": "_1EAygSCvc8a",
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "outputId": "db0dfd5f-4b1a-4c61-cb86-a598538dc2ae"
      },
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "\n",
            "Generating a 2-way \"venn\" diagram. Please wait...\n",
            "\n",
            "\n",
            "Done! Please check your results @ venn1. \n",
            "Thank you for using Intervene!\n",
            "\n"
          ]
        }
      ],
      "source": [
        "!intervene venn -i macs2/ENCFF052DPD_peaks.broadPeak ENCFF379KHF.bed -o venn1 --filenames"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "7qqn3n2cRdkt",
        "outputId": "ebbae105-0c8a-472c-fe11-2ff5972a667f"
      },
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "\n",
            "Generating a 2-way \"venn\" diagram. Please wait...\n",
            "\n",
            "\n",
            "Done! Please check your results @ venn2. \n",
            "Thank you for using Intervene!\n",
            "\n"
          ]
        }
      ],
      "source": [
        "!intervene venn -i ENCFF379KHF.bed macs2/ENCFF052DPD_peaks.broadPeak -o venn2 --filenames"
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "!intervene venn -i macs2/ENCFF726ZWD_peaks.broadPeak ENCFF379KHF.bed -o venn3 --filenames"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "iZMpe8MBiIoH",
        "outputId": "c26b77b7-747a-4439-b49f-2913e07d2b21"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "\n",
            "Generating a 2-way \"venn\" diagram. Please wait...\n",
            "\n",
            "\n",
            "Done! Please check your results @ venn3. \n",
            "Thank you for using Intervene!\n",
            "\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "!intervene venn -i ENCFF379KHF.bed macs2/ENCFF726ZWD_peaks.broadPeak -o venn4 --filenames"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "lT8If7HXiI9B",
        "outputId": "cd78ebd3-32a8-4291-fda8-d51be3dffb0d"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "\n",
            "Generating a 2-way \"venn\" diagram. Please wait...\n",
            "\n",
            "\n",
            "Done! Please check your results @ venn4. \n",
            "Thank you for using Intervene!\n",
            "\n"
          ]
        }
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "w_TpdMixmsdY"
      },
      "source": [
        "# Бонусная часть\n",
        "\n",
        "станет доступно с 25.02\n",
        "\n",
        "Планируется получение графиков ngs.plot в функциях ниже. Работает долго. Результат в result.png."
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "uVvHCgj45XQ0"
      },
      "source": [
        "Ищем .bam file\n",
        "https://www.encodeproject.org/experiments/ENCSR000AKC/"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "id": "LYl2b-HpjgOR"
      },
      "outputs": [],
      "source": [
        "!wget https://www.encodeproject.org/files/ENCFF882PRP/@@download/ENCFF882PRP.bam"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "id": "DTj12rByiWy5"
      },
      "outputs": [],
      "source": [
        "!pip install imgaug==0.2.5"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "id": "KhYvZAhYiWtl"
      },
      "outputs": [],
      "source": [
        "!pip install -q deeptools"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "id": "3wNbD7dgp1el"
      },
      "outputs": [],
      "source": [
        "!conda install -c bioconda bedtools"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "id": "IIGjc9EEdiV4"
      },
      "outputs": [],
      "source": [
        "!apt-get install bedtools"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "id": "kkjP1kvZiWmH"
      },
      "outputs": [],
      "source": [
        "!bedtools bamtobed -i ENCFF882PRP.bam > ENCFF882PRP.bed"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "ZM5iNuKKvpu1"
      },
      "source": [
        "Ищем .bigWig file связанный с выбранным .bam"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "id": "DAy5E9OllNda"
      },
      "outputs": [],
      "source": [
        "!wget https://www.encodeproject.org/files/ENCFF258KTL/@@download/ENCFF258KTL.bigWig"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "id": "iB56pWZ7iWrZ"
      },
      "outputs": [],
      "source": [
        "!computeMatrix scale-regions \\\n",
        " -S ENCFF258KTL.bigWig \\\n",
        " -R ENCFF882PRP.bed \\\n",
        " -bs 50 \\\n",
        " -m 10000 -b 3000 -a 3000 \\\n",
        " -out matrix.tab.gz \\\n",
        " --skipZeros \\\n",
        " --missingDataAsZero"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "id": "kspGb93Q5X6e"
      },
      "outputs": [],
      "source": [
        "!plotHeatmap \\\n",
        " -m matrix.tab.gz \\\n",
        " -out result.png \\\n",
        " --colorMap YlGnBu \\\n",
        " --regionsLabel 'genes' \\\n",
        " --heatmapHeight 15 \\\n",
        " --plotTitle 'ngs plot' &"
      ]
    },
    {
      "cell_type": "code",
      "source": [
        ""
      ],
      "metadata": {
        "id": "YLcQ3zkHfQCY"
      },
      "execution_count": null,
      "outputs": []
    }
  ],
  "metadata": {
    "colab": {
      "collapsed_sections": [],
      "name": "hse_hw2_chip.ipynb",
      "provenance": [],
      "include_colab_link": true
    },
    "kernelspec": {
      "display_name": "Python 3",
      "name": "python3"
    },
    "language_info": {
      "name": "python"
    }
  },
  "nbformat": 4,
  "nbformat_minor": 0
}